{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a6480aea-dff5-4dfd-a311-37620295657d",
   "metadata": {},
   "outputs": [],
   "source": [
    "#download samples: /BSSE_TREUTLEIN/TREUTLEIN/P2794_HSIU-CHUAN/POOL-631"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "7e0c512b-e290-468c-8d00-9fa517cbf5ad",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"/links/groups/treutlein/SCRIPTS/sequencing/get_sequencing_results.py\", line 6, in <module>\n",
      "    import git\n",
      "ModuleNotFoundError: No module named 'git'\n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "python /links/groups/treutlein/SCRIPTS/sequencing/get_sequencing_results.py -h"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "05e7cfd1-e266-4406-ab53-f69595c8fd2c",
   "metadata": {},
   "outputs": [],
   "source": [
    "python /links/groups/treutlein/SCRIPTS/sequencing/get_sequencing_results.py /BSSE_TREUTLEIN/POOL-631"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "a6e346e5-a8dc-4f36-8946-d9c3582e9163",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(bulk_seq) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "mamba activate bulk_seq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "813ec46a-f361-43a3-a3d6-b231d66010bc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/nas/groups/treutlein/USERS/jjans/mambaforge/envs/bulk_seq/bin/fastp\n",
      "(bulk_seq) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "which fastp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "8445948a-9b4e-46fc-b2b0-a6704bf7a31c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/nas/groups/treutlein/USERS/jjans/mambaforge/envs/bulk_seq/bin/STAR\n",
      "(bulk_seq) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "which STAR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "f5f7c975-a443-4913-a35d-a30954dabf07",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_24/iGABA_pre_24_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_24/iGABA_pre_24_S2_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "(bulk_seq) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "ls /links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_24/*R1*.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "a19bf4d6-696c-4636-a5fe-855fff38b873",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;32mUntitled.ipynb\u001b[0m  \u001b[01;34miGABA_pre_24_S1_L001_R1_001\u001b[0m  \u001b[01;34miGABA_pre_24_S2_L002_R1_001\u001b[0m\n",
      "(bulk_seq) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "473d2b49-d819-4322-88d2-0f342600e61f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;32mUntitled.ipynb\u001b[0m  \u001b[01;34miGABA_pre_24\u001b[0m\n",
      "(bulk_seq) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "04a6627d-7af4-4d07-b65f-f2bf3d589ee2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_24/iGABA_pre_24_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_24/iGABA_pre_24_S2_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "(bulk_seq) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "ls /links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_24/*R1*.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "83c992d9-8fca-4145-802d-5142f36b140f",
   "metadata": {},
   "outputs": [],
   "source": [
    "export MAXCORES=8\n",
    "\n",
    "ls /links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_24/*R1*.fastq.gz | parallel -j 15 --plus --rpl '{//} s:_S[0-9]+_L[0-9]+_R1_001.fastq.gz::;s:.*/::' \"\n",
    "    mkdir -p {/}\n",
    "    cd {/}\n",
    "    echo {}\n",
    "    \n",
    "    fastp -i {} -I {= s/_R1_/_R2_/ =} -o {/}/{/.}.clean_R1.fastq.gz -O {/}/{/.}.clean_R2.fastq.gz\n",
    "\" ::: /links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_24/*R1*.fastq.gz\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "a2170121-944f-4565-b017-574725681beb",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(bulk_seq) \n",
      "Academic tradition requires you to cite works you base your article on.\n",
      "If you use programs that use GNU Parallel to process data for an article in a\n",
      "scientific publication, please cite:\n",
      "\n",
      "  Tange, O. (2024, May 22). GNU Parallel 20240522 ('Tbilisi').\n",
      "  Zenodo. https://doi.org/10.5281/zenodo.11247979\n",
      "\n",
      "This helps funding further development; AND IT WON'T COST YOU A CENT.\n",
      "If you pay 10000 EUR you should feel free to use GNU Parallel without citing.\n",
      "\n",
      "More about funding GNU Parallel and the citation notice:\n",
      "https://www.gnu.org/software/parallel/parallel_design.html#citation-notice\n",
      "\n",
      "To silence this citation notice: run 'parallel --citation' once.\n",
      "\n",
      "Come on: You have run parallel 14 times. Isn't it about time \n",
      "you run 'parallel --citation' once to silence the citation notice?\n",
      "\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre_24\n",
      "/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_24/iGABA_pre_24_S1_L001_R1_001.fastq.gz\n",
      "iGABA_pre_24_S1_L001_R1_001\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre_24\n",
      "/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_24/iGABA_pre_24_S2_L002_R1_001.fastq.gz\n",
      "iGABA_pre_24_S2_L002_R1_001\n",
      "(bulk_seq) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "export MAXCORES=8\n",
    "ls /links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_24/*R1*.fastq.gz | parallel -j 15 --plus --rpl '{/..L} s:.fastq.gz::;s:.*/::;' --rpl '{/..} s:.fastq.gz::;s:.*/::;s:_S.*::' \"\n",
    "        mkdir -p {/..}\n",
    "        cd {/..}\n",
    "        echo {}\n",
    "        echo {/..L}\n",
    "\""
   ]
  },
  {
   "cell_type": "raw",
   "id": "daad0541-313a-4d06-9302-c5fbc7dcf1e8",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "source": [
    "#OLD\n",
    "export MAXCORES=8\n",
    "\n",
    "ls /links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_24/*R1*.fastq.gz | parallel -j 15 --plus --rpl '{//} s:_S[0-9]+_L[0-9]+_R1_001.fastq.gz::;s:.*/::' \"\n",
    "    mkdir -p {/}\n",
    "    cd {/}\n",
    "    echo {}\n",
    "    \n",
    "    fastp -i {} -I {= s/_R1_/_R2_/ =} -o {/.}_R1.clean.fastq.gz -O {= s/_R1_/_R2_/;s:.*/:: =}_R2.clean.fastq.gz\n",
    "\" ::: /links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_24/*R1*.fastq.gz\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "974c9a02-a772-4579-aadd-30b800272c4b",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(bulk_seq) \n",
      "(bulk_seq) \n",
      "Academic tradition requires you to cite works you base your article on.\n",
      "If you use programs that use GNU Parallel to process data for an article in a\n",
      "scientific publication, please cite:\n",
      "\n",
      "  Tange, O. (2024, May 22). GNU Parallel 20240522 ('Tbilisi').\n",
      "  Zenodo. https://doi.org/10.5281/zenodo.11247979\n",
      "\n",
      "This helps funding further development; AND IT WON'T COST YOU A CENT.\n",
      "If you pay 10000 EUR you should feel free to use GNU Parallel without citing.\n",
      "\n",
      "More about funding GNU Parallel and the citation notice:\n",
      "https://www.gnu.org/software/parallel/parallel_design.html#citation-notice\n",
      "\n",
      "To silence this citation notice: run 'parallel --citation' once.\n",
      "\n",
      "Come on: You have run parallel 22 times. Isn't it about time \n",
      "you run 'parallel --citation' once to silence the citation notice?\n",
      "\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre_24\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre_24\n",
      "(bulk_seq) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "export MAXCORES=8\n",
    "export STARGEN=/local1/sequencing/PUBLIC_DATA/genomes/refdata-cellranger-arc-GRCh38-2020-A-2.0.0/star/\n",
    "ls /links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_24/*R1*.fastq.gz | parallel -j 15 --plus --rpl '{/..L} s:.fastq.gz::;s:.*/::;s:_R.*::' --rpl '{/..} s:.fastq.gz::;s:.*/::;s:_S.*::' \"\n",
    "        mkdir -p {/..}\n",
    "        cd {/..}\n",
    "        \n",
    "        mkdir -p 00.CLEAN_FQ\n",
    "        fastp -i {} -I {=s/R1/R2/=} -o 00.CLEAN_FQ/{/..L}_R1.clean.fastq.gz -O 00.CLEAN_FQ/{/..L}_R2.clean.fastq.gz -h 00.CLEAN_FQ/{/..L}_fastp.html -j 00.CLEAN_FQ/{/..L}_fastp.json &>> {/..L}_fastp.log\n",
    "        \n",
    "        cat 00.CLEAN_FQ/*_R1.clean.fastq.gz > 00.CLEAN_FQ/{/..}_R1.merged.fastq.gz\n",
    "        cat 00.CLEAN_FQ/*_R2.clean.fastq.gz > 00.CLEAN_FQ/{/..}_R2.merged.fastq.gz\n",
    "       \n",
    "\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "e4617140-792d-48bf-8599-7242a92b8cae",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-rwxrwx--- 1 jjanssens@d.ethz.ch bsse-treutlein@d.ethz.ch 69M May 28 15:57 \u001b[0m\u001b[01;32miGABA_pre_24/00.CLEAN_FQ/iGABA_pre_24_R1.merged.fastq.gz\u001b[0m\n",
      "-rwxrwx--- 1 jjanssens@d.ethz.ch bsse-treutlein@d.ethz.ch 71M May 28 15:57 \u001b[01;32miGABA_pre_24/00.CLEAN_FQ/iGABA_pre_24_R2.merged.fastq.gz\u001b[0m\n",
      "-rwxrwx--- 1 jjanssens@d.ethz.ch bsse-treutlein@d.ethz.ch 34M May 28 15:57 \u001b[01;32miGABA_pre_24/00.CLEAN_FQ/iGABA_pre_24_S1_L001_R1.clean.fastq.gz\u001b[0m\n",
      "-rwxrwx--- 1 jjanssens@d.ethz.ch bsse-treutlein@d.ethz.ch 35M May 28 15:57 \u001b[01;32miGABA_pre_24/00.CLEAN_FQ/iGABA_pre_24_S1_L001_R2.clean.fastq.gz\u001b[0m\n",
      "-rwxrwx--- 1 jjanssens@d.ethz.ch bsse-treutlein@d.ethz.ch 35M May 28 15:57 \u001b[01;32miGABA_pre_24/00.CLEAN_FQ/iGABA_pre_24_S2_L002_R1.clean.fastq.gz\u001b[0m\n",
      "-rwxrwx--- 1 jjanssens@d.ethz.ch bsse-treutlein@d.ethz.ch 36M May 28 15:57 \u001b[01;32miGABA_pre_24/00.CLEAN_FQ/iGABA_pre_24_S2_L002_R2.clean.fastq.gz\u001b[0m\n",
      "(bulk_seq) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "ls -lah iGABA_pre_24/00.CLEAN_FQ/*.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "816aa6ba-9957-4454-a108-d2611bd00b83",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;32mHomo_sapiens.GRCh38.111.chr.gtf.gz\u001b[0m\n",
      "\u001b[01;32mHomo_sapiens.GRCh38.111.filtered.gtf\u001b[0m\n",
      "\u001b[01;32mHomo_sapiens.GRCh38.111.gtf.gz\u001b[0m\n",
      "\u001b[01;32mHomo_sapiens.GRCh38.dna.toplevel.fa\u001b[0m\n",
      "\u001b[01;32mHomo_sapiens.GRCh38.dna_rm.primary_assembly.fa.gz\u001b[0m\n",
      "\u001b[01;32mreadme.txt\u001b[0m\n",
      "(bulk_seq) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "ls /links/groups/treutlein/USERS/jjans/resources/genomes/hsapiens/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "faeb55d8-3744-4663-a35f-e513640d0f07",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(bulk_seq) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "mkdir /links/groups/treutlein/USERS/jjans/resources/genomes/STAR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "b934a593-46e8-40d8-84c4-37c8ccfeb8a4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(bulk_seq) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "gunzip -c /links/groups/treutlein/USERS/jjans/resources/genomes/hsapiens/Homo_sapiens.GRCh38.111.chr.gtf.gz > /links/groups/treutlein/USERS/jjans/resources/genomes/hsapiens/Homo_sapiens.GRCh38.111.chr.gtf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "757c4515-7f79-43e5-acc6-bb40a701e4a0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-rwxrwx--- 1 jjanssens@d.ethz.ch bsse-treutlein@d.ethz.ch 1.4G May 28 16:21 \u001b[0m\u001b[01;32m/links/groups/treutlein/USERS/jjans/resources/genomes/hsapiens/Homo_sapiens.GRCh38.111.chr.gtf\u001b[0m\u001b[K\n",
      "(bulk_seq) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "ls -lah /links/groups/treutlein/USERS/jjans/resources/genomes/hsapiens/Homo_sapiens.GRCh38.111.chr.gtf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "637e17bf-3200-46a8-a8d0-3895d21dea19",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(bulk_seq) \n",
      "(bulk_seq) \n",
      "(bulk_seq) \n",
      "(bulk_seq) \n",
      "(bulk_seq) \n",
      "(bulk_seq) \n",
      "(bulk_seq) \n",
      "(bulk_seq) \n",
      "(bulk_seq) \n",
      "\t/nas/groups/treutlein/USERS/jjans/mambaforge/envs/bulk_seq/bin/STAR-avx2 --runThreadN 20 --runMode genomeGenerate --genomeDir /links/groups/treutlein/USERS/jjans/resources/genomes/STAR/Homo_sapiens.GRCh38 --genomeFastaFiles /links/groups/treutlein/USERS/jjans/resources/genomes/hsapiens/Homo_sapiens.GRCh38.dna.toplevel.fa --sjdbGTFfile /links/groups/treutlein/USERS/jjans/resources/genomes/hsapiens/Homo_sapiens.GRCh38.111.chr.gtf --sjdbOverhang 100\n",
      "\tSTAR version: 2.7.11b   compiled: 2024-03-19T08:38:59+0000 :/opt/conda/conda-bld/star_1710837244939/work/source\n",
      "May 28 16:21:15 ..... started STAR run\n",
      "May 28 16:21:15 ... starting to generate Genome files\n",
      "May 28 16:21:51 ..... processing annotations GTF\n",
      "May 28 16:22:17 ... starting to sort Suffix Array. This may take a long time...\n",
      "May 28 16:22:30 ... sorting Suffix Array chunks and saving them to disk...\n",
      "May 28 16:45:59 ... loading chunks from disk, packing SA...\n",
      "May 28 16:47:15 ... finished generating suffix array\n",
      "May 28 16:47:15 ... generating Suffix Array index\n",
      "May 28 16:51:58 ... completed Suffix Array index\n",
      "May 28 16:51:59 ..... inserting junctions into the genome indices\n",
      "May 28 16:55:09 ... writing Genome to disk ...\n",
      "May 28 16:55:18 ... writing Suffix Array to disk ...\n",
      "May 28 16:56:21 ... writing SAindex to disk\n",
      "May 28 16:56:29 ..... finished successfully\n",
      "(bulk_seq) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "# Set variables for your files and directories\n",
    "GENOME_FASTA=\"/links/groups/treutlein/USERS/jjans/resources/genomes/hsapiens/Homo_sapiens.GRCh38.dna.toplevel.fa\"\n",
    "GTF_FILE=\"/links/groups/treutlein/USERS/jjans/resources/genomes/hsapiens/Homo_sapiens.GRCh38.111.chr.gtf\"\n",
    "STAR_INDEX_DIR=\"/links/groups/treutlein/USERS/jjans/resources/genomes/STAR/Homo_sapiens.GRCh38\"\n",
    "\n",
    "# Create the output directory if it doesn't exist\n",
    "mkdir -p $STAR_INDEX_DIR\n",
    "\n",
    "# Run STAR to generate the genome index\n",
    "STAR --runThreadN 20 \\\n",
    "     --runMode genomeGenerate \\\n",
    "     --genomeDir $STAR_INDEX_DIR \\\n",
    "     --genomeFastaFiles $GENOME_FASTA \\\n",
    "     --sjdbGTFfile $GTF_FILE \\\n",
    "     --sjdbOverhang 100\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0c5d6d80-da71-4b41-a038-1309b1b0b062",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "07ea310a-8ee0-4ffe-9187-81c9a3cf07bb",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;32mGenome\u001b[0m   \u001b[01;32mchrLength.txt\u001b[0m      \u001b[01;32mexonGeTrInfo.tab\u001b[0m      \u001b[01;32msjdbInfo.txt\u001b[0m\n",
      "\u001b[01;32mLog.out\u001b[0m  \u001b[01;32mchrName.txt\u001b[0m        \u001b[01;32mexonInfo.tab\u001b[0m          \u001b[01;32msjdbList.fromGTF.out.tab\u001b[0m\n",
      "\u001b[01;32mSA\u001b[0m       \u001b[01;32mchrNameLength.txt\u001b[0m  \u001b[01;32mgeneInfo.tab\u001b[0m          \u001b[01;32msjdbList.out.tab\u001b[0m\n",
      "\u001b[01;32mSAindex\u001b[0m  \u001b[01;32mchrStart.txt\u001b[0m       \u001b[01;32mgenomeParameters.txt\u001b[0m  \u001b[01;32mtranscriptInfo.tab\u001b[0m\n",
      "(bulk_seq) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "ls /links/groups/treutlein/USERS/jjans/resources/genomes/STAR/Homo_sapiens.GRCh38"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "9c7f6508-cea3-4d05-8833-0f1a4a3d3e2a",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(bulk_seq) \n",
      "(bulk_seq) \n",
      "(bulk_seq) \n",
      "(bulk_seq) \n",
      "(bulk_seq) \n",
      "(bulk_seq) \n",
      "(bulk_seq) \n",
      "Academic tradition requires you to cite works you base your article on.\n",
      "If you use programs that use GNU Parallel to process data for an article in a\n",
      "scientific publication, please cite:\n",
      "\n",
      "  Tange, O. (2024, May 22). GNU Parallel 20240522 ('Tbilisi').\n",
      "  Zenodo. https://doi.org/10.5281/zenodo.11247979\n",
      "\n",
      "This helps funding further development; AND IT WON'T COST YOU A CENT.\n",
      "If you pay 10000 EUR you should feel free to use GNU Parallel without citing.\n",
      "\n",
      "More about funding GNU Parallel and the citation notice:\n",
      "https://www.gnu.org/software/parallel/parallel_design.html#citation-notice\n",
      "\n",
      "To silence this citation notice: run 'parallel --citation' once.\n",
      "\n",
      "Come on: You have run parallel 27 times. Isn't it about time \n",
      "you run 'parallel --citation' once to silence the citation notice?\n",
      "\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre_24\n",
      "(bulk_seq) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "export MAXCORES=8\n",
    "export STARGEN=/local1/sequencing/PUBLIC_DATA/genomes/refdata-cellranger-arc-GRCh38-2020-A-2.0.0/star/\n",
    "\n",
    "export STARGEN=/links/groups/treutlein/USERS/jjans/software/cellranger/refdata-gex-GRCh38-2020-A/star/\n",
    "\n",
    "export STARGEN=/links/groups/treutlein/USERS/jjans/resources/genomes/STAR/Homo_sapiens.GRCh38\n",
    "\n",
    "ls /links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_24/*S1*R1*.fastq.gz | parallel -j 15 --plus --rpl '{/..L} s:.fastq.gz::;s:.*/::;' --rpl '{/..} s:.fastq.gz::;s:.*/::;s:_S.*::' \"\n",
    "        cd {/..}\n",
    "        \n",
    "        mkdir -p 10.MAPPING\n",
    "        \n",
    "        STAR \\\n",
    "        --runThreadN ${MAXCORES} \\\n",
    "        --genomeDir ${STARGEN} \\\n",
    "        --readFilesIn 00.CLEAN_FQ/*merged*.fastq.gz \\\n",
    "        --readFilesCommand zcat \\\n",
    "        --outFileNamePrefix 10.MAPPING/{/..}_STAR_ \\\n",
    "        --outSAMtype BAM SortedByCoordinate \\\n",
    "        --outWigType bedGraph \\\n",
    "        --outWigStrand Unstranded \\\n",
    "        --outWigNorm RPM \\\n",
    "        --outReadsUnmapped Fastx \\\n",
    "        --quantMode GeneCounts \\\n",
    "        &>> {/..}_STAR.log\n",
    "\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "2084b870-3d66-414e-8bbc-8991af62f9df",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_1/iGABA_pre_1_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_10/iGABA_pre_10_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_11/iGABA_pre_11_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_12/iGABA_pre_12_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_13/iGABA_pre_13_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_14/iGABA_pre_14_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_15/iGABA_pre_15_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_16/iGABA_pre_16_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_17/iGABA_pre_17_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_18/iGABA_pre_18_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_19/iGABA_pre_19_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_2/iGABA_pre_2_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_20/iGABA_pre_20_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_21/iGABA_pre_21_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_22/iGABA_pre_22_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_23/iGABA_pre_23_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_24/iGABA_pre_24_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_25/iGABA_pre_25_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_26/iGABA_pre_26_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_27/iGABA_pre_27_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_28/iGABA_pre_28_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_29/iGABA_pre_29_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_3/iGABA_pre_3_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_30/iGABA_pre_30_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_31/iGABA_pre_31_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_32/iGABA_pre_32_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_33/iGABA_pre_33_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_34/iGABA_pre_34_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_35/iGABA_pre_35_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_36/iGABA_pre_36_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_37/iGABA_pre_37_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_38/iGABA_pre_38_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_39/iGABA_pre_39_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_4/iGABA_pre_4_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_40/iGABA_pre_40_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_41/iGABA_pre_41_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_42/iGABA_pre_42_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_43/iGABA_pre_43_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_44/iGABA_pre_44_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_45/iGABA_pre_45_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_46/iGABA_pre_46_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_47/iGABA_pre_47_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_48/iGABA_pre_48_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_49/iGABA_pre_49_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_5/iGABA_pre_5_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_50/iGABA_pre_50_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_51/iGABA_pre_51_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_52/iGABA_pre_52_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_53/iGABA_pre_53_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_54/iGABA_pre_54_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_55/iGABA_pre_55_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_56/iGABA_pre_56_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_57/iGABA_pre_57_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_58/iGABA_pre_58_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_59/iGABA_pre_59_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_6/iGABA_pre_6_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_60/iGABA_pre_60_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_61/iGABA_pre_61_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_62/iGABA_pre_62_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_63/iGABA_pre_63_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_64/iGABA_pre_64_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_65/iGABA_pre_65_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_66/iGABA_pre_66_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_67/iGABA_pre_67_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_68/iGABA_pre_68_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_69/iGABA_pre_69_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_7/iGABA_pre_7_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_70/iGABA_pre_70_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_71/iGABA_pre_71_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_72/iGABA_pre_72_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_73/iGABA_pre_73_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_74/iGABA_pre_74_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_75/iGABA_pre_75_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_76/iGABA_pre_76_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_77/iGABA_pre_77_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_78/iGABA_pre_78_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_79/iGABA_pre_79_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_8/iGABA_pre_8_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_80/iGABA_pre_80_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_81/iGABA_pre_81_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_82/iGABA_pre_82_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_83/iGABA_pre_83_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_84/iGABA_pre_84_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_85/iGABA_pre_85_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_86/iGABA_pre_86_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_87/iGABA_pre_87_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_88/iGABA_pre_88_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_89/iGABA_pre_89_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_9/iGABA_pre_9_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_90/iGABA_pre_90_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_91/iGABA_pre_91_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_92/iGABA_pre_92_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_93/iGABA_pre_93_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_94/iGABA_pre_94_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_95/iGABA_pre_95_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_96/iGABA_pre_96_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "(bulk_seq) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "ls /links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_*/*S1*R1*.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "id": "06a2593a-5755-47a9-b484-7e6142943940",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(bulk_seq) \n",
      "(bulk_seq) \n",
      "(bulk_seq) \n",
      "(bulk_seq) \n",
      "(bulk_seq) \n",
      "(bulk_seq) \n",
      "(bulk_seq) \n",
      "Academic tradition requires you to cite works you base your article on.\n",
      "If you use programs that use GNU Parallel to process data for an article in a\n",
      "scientific publication, please cite:\n",
      "\n",
      "  Tange, O. (2024, May 22). GNU Parallel 20240522 ('Tbilisi').\n",
      "  Zenodo. https://doi.org/10.5281/zenodo.11247979\n",
      "\n",
      "This helps funding further development; AND IT WON'T COST YOU A CENT.\n",
      "If you pay 10000 EUR you should feel free to use GNU Parallel without citing.\n",
      "\n",
      "More about funding GNU Parallel and the citation notice:\n",
      "https://www.gnu.org/software/parallel/parallel_design.html#citation-notice\n",
      "\n",
      "To silence this citation notice: run 'parallel --citation' once.\n",
      "\n",
      "Come on: You have run parallel 28 times. Isn't it about time \n",
      "you run 'parallel --citation' once to silence the citation notice?\n",
      "\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_20\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_14\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_16\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_13\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_15\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_12\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_10\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_17\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_18\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_22\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_11\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_21\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_2\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_19\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_1\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_30\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_24\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_29\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_31\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_28\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_23\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_26\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_25\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_27\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_32\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_3\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_33\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_35\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_37\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_36\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_34\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_38\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_45\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_41\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_42\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_40\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_39\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_4\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_43\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_44\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_46\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_47\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_48\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_50\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_49\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_52\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_5\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_51\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_57\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_58\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_53\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_56\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_55\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_54\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_59\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_6\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_62\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_61\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_60\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_65\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_66\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_68\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_64\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_63\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_67\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_69\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_74\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_71\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_73\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_70\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_72\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_7\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_75\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_77\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_78\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_76\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_8\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_79\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_80\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_81\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_82\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_83\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_84\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_85\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_86\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_87\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_89\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_88\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_9\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_90\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_91\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_95\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_96\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_94\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_93\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/iGABA_pre/iGABA_pre_92\n",
      "(bulk_seq) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "export MAXCORES=8\n",
    "export STARGEN=/local1/sequencing/PUBLIC_DATA/genomes/refdata-cellranger-arc-GRCh38-2020-A-2.0.0/star/\n",
    "\n",
    "export STARGEN=/links/groups/treutlein/USERS/jjans/software/cellranger/refdata-gex-GRCh38-2020-A/star/\n",
    "\n",
    "export STARGEN=/links/groups/treutlein/USERS/jjans/resources/genomes/STAR/Homo_sapiens.GRCh38\n",
    "\n",
    "ls /links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_*/*S1*R1*.fastq.gz | parallel -j 15 --plus --rpl '{/..L} s:.fastq.gz::;s:.*/::;' --rpl '{/..} s:.fastq.gz::;s:.*/::;s:_S.*::' \"\n",
    "        cd iGABA_pre/{/..}\n",
    "        \n",
    "        mkdir -p 10.MAPPING\n",
    "        \n",
    "        STAR \\\n",
    "        --runThreadN ${MAXCORES} \\\n",
    "        --genomeDir ${STARGEN} \\\n",
    "        --readFilesIn 00.CLEAN_FQ/*merged*.fastq.gz \\\n",
    "        --readFilesCommand zcat \\\n",
    "        --outFileNamePrefix 10.MAPPING/{/..}_STAR_ \\\n",
    "        --outSAMtype BAM SortedByCoordinate \\\n",
    "        --outWigType bedGraph \\\n",
    "        --outWigStrand Unstranded \\\n",
    "        --outWigNorm RPM \\\n",
    "        --outReadsUnmapped Fastx \\\n",
    "        --quantMode GeneCounts \\\n",
    "        &>> {/..}_STAR.log\n",
    "\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "e361fc83-e73c-446a-98e3-91cb6c735aee",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d10_sample_1/Stability_d10_sample_1_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d10_sample_10/Stability_d10_sample_10_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d10_sample_11/Stability_d10_sample_11_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d10_sample_12/Stability_d10_sample_12_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d10_sample_2/Stability_d10_sample_2_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d10_sample_3/Stability_d10_sample_3_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d10_sample_4/Stability_d10_sample_4_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d10_sample_5/Stability_d10_sample_5_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d10_sample_6/Stability_d10_sample_6_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d10_sample_7/Stability_d10_sample_7_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d10_sample_8/Stability_d10_sample_8_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d10_sample_9/Stability_d10_sample_9_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "(bulk_seq) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "ls /links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d10*/*S1*R1*.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "bfc046b3-1bed-47dc-9374-5fbfda1b6911",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(bulk_seq) \n",
      "(bulk_seq) \n",
      "(bulk_seq) \n",
      "Academic tradition requires you to cite works you base your article on.\n",
      "If you use programs that use GNU Parallel to process data for an article in a\n",
      "scientific publication, please cite:\n",
      "\n",
      "  Tange, O. (2024, May 22). GNU Parallel 20240522 ('Tbilisi').\n",
      "  Zenodo. https://doi.org/10.5281/zenodo.11247979\n",
      "\n",
      "This helps funding further development; AND IT WON'T COST YOU A CENT.\n",
      "If you pay 10000 EUR you should feel free to use GNU Parallel without citing.\n",
      "\n",
      "More about funding GNU Parallel and the citation notice:\n",
      "https://www.gnu.org/software/parallel/parallel_design.html#citation-notice\n",
      "\n",
      "To silence this citation notice: run 'parallel --citation' once.\n",
      "\n",
      "Come on: You have run parallel 29 times. Isn't it about time \n",
      "you run 'parallel --citation' once to silence the citation notice?\n",
      "\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_409B2/Reproducibility_409B2_1_11\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_409B2/Reproducibility_409B2_1_10\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_409B2/Reproducibility_409B2_1_12\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_409B2/Reproducibility_409B2_1_9\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_409B2/Reproducibility_409B2_2_10\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_409B2/Reproducibility_409B2_2_11\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_409B2/Reproducibility_409B2_2_1\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_409B2/Reproducibility_409B2_1_5\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_409B2/Reproducibility_409B2_1_3\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_409B2/Reproducibility_409B2_1_4\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_409B2/Reproducibility_409B2_1_8\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_409B2/Reproducibility_409B2_1_6\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_409B2/Reproducibility_409B2_1_1\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_409B2/Reproducibility_409B2_1_2\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_409B2/Reproducibility_409B2_1_7\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_409B2/Reproducibility_409B2_2_12\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_409B2/Reproducibility_409B2_2_2\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_409B2/Reproducibility_409B2_2_4\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_409B2/Reproducibility_409B2_2_5\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_409B2/Reproducibility_409B2_2_3\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_409B2/Reproducibility_409B2_2_6\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_409B2/Reproducibility_409B2_3_10\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_409B2/Reproducibility_409B2_2_8\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_409B2/Reproducibility_409B2_3_12\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_409B2/Reproducibility_409B2_3_1\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_409B2/Reproducibility_409B2_2_7\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_409B2/Reproducibility_409B2_3_11\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_409B2/Reproducibility_409B2_3_2\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_409B2/Reproducibility_409B2_2_9\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_409B2/Reproducibility_409B2_3_3\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_409B2/Reproducibility_409B2_3_4\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_409B2/Reproducibility_409B2_3_9\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_409B2/Reproducibility_409B2_3_5\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_409B2/Reproducibility_409B2_3_8\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_409B2/Reproducibility_409B2_3_7\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_409B2/Reproducibility_409B2_3_6\n",
      "(bulk_seq) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "export MAXCORES=8\n",
    "export STARGEN=/links/groups/treutlein/USERS/jjans/resources/genomes/STAR/Homo_sapiens.GRCh38\n",
    "\n",
    "ls /links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Reproducibility_409B2*/*S1*R1*.fastq.gz | parallel -j 15 --plus --rpl '{/..L} s:.fastq.gz::;s:.*/::;' --rpl '{/..} s:.fastq.gz::;s:.*/::;s:_S.*::' \"\n",
    "        cd Reproducibility_409B2/{/..}\n",
    "        \n",
    "        mkdir -p 10.MAPPING\n",
    "        \n",
    "        STAR \\\n",
    "        --runThreadN ${MAXCORES} \\\n",
    "        --genomeDir ${STARGEN} \\\n",
    "        --readFilesIn 00.CLEAN_FQ/*merged*.fastq.gz \\\n",
    "        --readFilesCommand zcat \\\n",
    "        --outFileNamePrefix 10.MAPPING/{/..}_STAR_ \\\n",
    "        --outSAMtype BAM SortedByCoordinate \\\n",
    "        --outWigType bedGraph \\\n",
    "        --outWigStrand Unstranded \\\n",
    "        --outWigNorm RPM \\\n",
    "        --outReadsUnmapped Fastx \\\n",
    "        --quantMode GeneCounts \\\n",
    "        &>> {/..}_STAR.log\n",
    "\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "id": "f97eb081-4780-4a1e-a073-61b1d6565bc6",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d10_sample_1/Stability_d10_sample_1_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d10_sample_10/Stability_d10_sample_10_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d10_sample_11/Stability_d10_sample_11_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d10_sample_12/Stability_d10_sample_12_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d10_sample_2/Stability_d10_sample_2_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d10_sample_3/Stability_d10_sample_3_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d10_sample_4/Stability_d10_sample_4_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d10_sample_5/Stability_d10_sample_5_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d10_sample_6/Stability_d10_sample_6_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d10_sample_7/Stability_d10_sample_7_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d10_sample_8/Stability_d10_sample_8_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d10_sample_9/Stability_d10_sample_9_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d21_sample_1/Stability_d21_sample_1_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d21_sample_10/Stability_d21_sample_10_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d21_sample_11/Stability_d21_sample_11_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d21_sample_12/Stability_d21_sample_12_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d21_sample_2/Stability_d21_sample_2_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d21_sample_3/Stability_d21_sample_3_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d21_sample_4/Stability_d21_sample_4_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d21_sample_5/Stability_d21_sample_5_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d21_sample_6/Stability_d21_sample_6_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d21_sample_7/Stability_d21_sample_7_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d21_sample_8/Stability_d21_sample_8_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d21_sample_9/Stability_d21_sample_9_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d28_sample_1/Stability_d28_sample_1_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d28_sample_10/Stability_d28_sample_10_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d28_sample_11/Stability_d28_sample_11_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d28_sample_12/Stability_d28_sample_12_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d28_sample_2/Stability_d28_sample_2_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d28_sample_3/Stability_d28_sample_3_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d28_sample_4/Stability_d28_sample_4_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d28_sample_5/Stability_d28_sample_5_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d28_sample_6/Stability_d28_sample_6_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d28_sample_7/Stability_d28_sample_7_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d28_sample_8/Stability_d28_sample_8_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d28_sample_9/Stability_d28_sample_9_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d35_sample_1/Stability_d35_sample_1_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d35_sample_10/Stability_d35_sample_10_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d35_sample_11/Stability_d35_sample_11_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d35_sample_12/Stability_d35_sample_12_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d35_sample_2/Stability_d35_sample_2_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d35_sample_3/Stability_d35_sample_3_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d35_sample_4/Stability_d35_sample_4_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d35_sample_5/Stability_d35_sample_5_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d35_sample_6/Stability_d35_sample_6_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d35_sample_7/Stability_d35_sample_7_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d35_sample_8/Stability_d35_sample_8_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d35_sample_9/Stability_d35_sample_9_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d42_sample_1/Stability_d42_sample_1_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d42_sample_10/Stability_d42_sample_10_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d42_sample_11/Stability_d42_sample_11_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d42_sample_12/Stability_d42_sample_12_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d42_sample_2/Stability_d42_sample_2_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d42_sample_3/Stability_d42_sample_3_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d42_sample_4/Stability_d42_sample_4_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d42_sample_5/Stability_d42_sample_5_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d42_sample_6/Stability_d42_sample_6_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d42_sample_7/Stability_d42_sample_7_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d42_sample_8/Stability_d42_sample_8_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d42_sample_9/Stability_d42_sample_9_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d49_sample_1/Stability_d49_sample_1_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d49_sample_10/Stability_d49_sample_10_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d49_sample_11/Stability_d49_sample_11_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d49_sample_12/Stability_d49_sample_12_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d49_sample_2/Stability_d49_sample_2_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d49_sample_3/Stability_d49_sample_3_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d49_sample_4/Stability_d49_sample_4_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d49_sample_5/Stability_d49_sample_5_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d49_sample_6/Stability_d49_sample_6_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d49_sample_7/Stability_d49_sample_7_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d49_sample_8/Stability_d49_sample_8_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability_d49_sample_9/Stability_d49_sample_9_S1_L002_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "(bulk_seq) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "ls /links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability*/*S1*R1*.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "id": "58550fe9-578a-4d51-a8a2-7222cc0e5f5f",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(bulk_seq) \n",
      "(bulk_seq) \n",
      "(bulk_seq) \n",
      "Academic tradition requires you to cite works you base your article on.\n",
      "If you use programs that use GNU Parallel to process data for an article in a\n",
      "scientific publication, please cite:\n",
      "\n",
      "  Tange, O. (2024, May 22). GNU Parallel 20240522 ('Tbilisi').\n",
      "  Zenodo. https://doi.org/10.5281/zenodo.11247979\n",
      "\n",
      "This helps funding further development; AND IT WON'T COST YOU A CENT.\n",
      "If you pay 10000 EUR you should feel free to use GNU Parallel without citing.\n",
      "\n",
      "More about funding GNU Parallel and the citation notice:\n",
      "https://www.gnu.org/software/parallel/parallel_design.html#citation-notice\n",
      "\n",
      "To silence this citation notice: run 'parallel --citation' once.\n",
      "\n",
      "Come on: You have run parallel 40 times. Isn't it about time \n",
      "you run 'parallel --citation' once to silence the citation notice?\n",
      "\n",
      "Stability_d10\n",
      "Stability_d10\n",
      "Stability_d10\n",
      "Stability_d10\n",
      "Stability_d10\n",
      "Stability_d10\n",
      "Stability_d10\n",
      "Stability_d10\n",
      "Stability_d10\n",
      "Stability_d10\n",
      "Stability_d10\n",
      "Stability_d10\n",
      "Stability_d21\n",
      "Stability_d21\n",
      "Stability_d21\n",
      "Stability_d21\n",
      "Stability_d21\n",
      "Stability_d21\n",
      "Stability_d21\n",
      "Stability_d21\n",
      "Stability_d21\n",
      "Stability_d21\n",
      "Stability_d21\n",
      "Stability_d21\n",
      "Stability_d28\n",
      "Stability_d28\n",
      "Stability_d28\n",
      "Stability_d28\n",
      "Stability_d28\n",
      "Stability_d28\n",
      "Stability_d28\n",
      "Stability_d28\n",
      "Stability_d28\n",
      "Stability_d28\n",
      "Stability_d28\n",
      "Stability_d28\n",
      "Stability_d35\n",
      "Stability_d35\n",
      "Stability_d35\n",
      "Stability_d35\n",
      "Stability_d35\n",
      "Stability_d35\n",
      "Stability_d35\n",
      "Stability_d35\n",
      "Stability_d35\n",
      "Stability_d35\n",
      "Stability_d35\n",
      "Stability_d35\n",
      "Stability_d42\n",
      "Stability_d42\n",
      "Stability_d42\n",
      "Stability_d42\n",
      "Stability_d42\n",
      "Stability_d42\n",
      "Stability_d42\n",
      "Stability_d42\n",
      "Stability_d42\n",
      "Stability_d42\n",
      "Stability_d42\n",
      "Stability_d42\n",
      "Stability_d49\n",
      "Stability_d49\n",
      "Stability_d49\n",
      "Stability_d49\n",
      "Stability_d49\n",
      "Stability_d49\n",
      "Stability_d49\n",
      "Stability_d49\n",
      "Stability_d49\n",
      "Stability_d49\n",
      "Stability_d49\n",
      "Stability_d49\n",
      "(bulk_seq) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "export MAXCORES=8\n",
    "export STARGEN=/links/groups/treutlein/USERS/jjans/resources/genomes/STAR/Homo_sapiens.GRCh38\n",
    "\n",
    "ls /links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability*/*S1*R1*.fastq.gz | parallel -j 15 --plus --rpl '{/..L} s:.fastq.gz::;s:.*/::;' --rpl '{/..} s:.fastq.gz::;s:.*/::;s:_S.*::' --rpl '{S} s:.fastq.gz::;s:.*/::;s:_sample.*::' \"\n",
    "    echo {S}\n",
    "\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a1850de7-3ce8-4e6f-bf48-486dda57b75a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(bulk_seq) \n",
      "(bulk_seq) \n",
      "(bulk_seq) \n",
      "Academic tradition requires you to cite works you base your article on.\n",
      "If you use programs that use GNU Parallel to process data for an article in a\n",
      "scientific publication, please cite:\n",
      "\n",
      "  Tange, O. (2024, May 22). GNU Parallel 20240522 ('Tbilisi').\n",
      "  Zenodo. https://doi.org/10.5281/zenodo.11247979\n",
      "\n",
      "This helps funding further development; AND IT WON'T COST YOU A CENT.\n",
      "If you pay 10000 EUR you should feel free to use GNU Parallel without citing.\n",
      "\n",
      "More about funding GNU Parallel and the citation notice:\n",
      "https://www.gnu.org/software/parallel/parallel_design.html#citation-notice\n",
      "\n",
      "To silence this citation notice: run 'parallel --citation' once.\n",
      "\n",
      "Come on: You have run parallel 41 times. Isn't it about time \n",
      "you run 'parallel --citation' once to silence the citation notice?\n",
      "\n"
     ]
    }
   ],
   "source": [
    "export MAXCORES=8\n",
    "export STARGEN=/links/groups/treutlein/USERS/jjans/resources/genomes/STAR/Homo_sapiens.GRCh38\n",
    "\n",
    "ls /links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Stability*/*S1*R1*.fastq.gz | parallel -j 15 --plus --rpl '{/..L} s:.fastq.gz::;s:.*/::;' --rpl '{/..} s:.fastq.gz::;s:.*/::;s:_S.*::' --rpl '{S} s:.fastq.gz::;s:.*/::;s:_sample.*::' \"\n",
    "        cd {S}/{/..}\n",
    "        \n",
    "        mkdir -p 10.MAPPING\n",
    "        \n",
    "        STAR \\\n",
    "        --runThreadN ${MAXCORES} \\\n",
    "        --genomeDir ${STARGEN} \\\n",
    "        --readFilesIn 00.CLEAN_FQ/*merged*.fastq.gz \\\n",
    "        --readFilesCommand zcat \\\n",
    "        --outFileNamePrefix 10.MAPPING/{/..}_STAR_ \\\n",
    "        --outSAMtype BAM SortedByCoordinate \\\n",
    "        --outWigType bedGraph \\\n",
    "        --outWigStrand Unstranded \\\n",
    "        --outWigNorm RPM \\\n",
    "        --outReadsUnmapped Fastx \\\n",
    "        --quantMode GeneCounts \\\n",
    "        &>> {/..}_STAR.log\n",
    "\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c60fde08-23ed-40f1-83eb-0d2c8df939ca",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "1a18fd5d-e00f-4bdf-85c3-fc9567eb48c0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(bulk_seq) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "mamba activate bulk_seq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "e6e43fb9-c467-4f18-afd6-2a90438c459c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;32mReproducibility_SC102A1_1_1_S1_L001_R1_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mReproducibility_SC102A1_1_1_S1_L001_R2_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mReproducibility_SC102A1_1_1_S2_L002_R1_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mReproducibility_SC102A1_1_1_S2_L002_R2_001.fastq.gz\u001b[0m\n",
      "(bulk_seq) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "ls /links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Reproducibility_SC102A1_1_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "296ea29a-4b63-4c72-9a6a-49ca4bf96402",
   "metadata": {},
   "outputs": [],
   "source": [
    "ls /links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Reproducibility_SC102A1_*/*S1*R1*.fastq.gz | parallel -j 15 --plus --rpl '{/..L} s:.fastq.gz::;s:.*/::;' --rpl '{/..} s:.fastq.gz::;s:.*/::;s:_S.*::' --rpl '{S} s:.fastq.gz::;s:.*/::;s:_sample.*::' "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "6d5f25fe-8159-46df-8b56-9d8fc0b7c7cb",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(bulk_seq) \n",
      "(bulk_seq) \n",
      "(bulk_seq) \n",
      "Academic tradition requires you to cite works you base your article on.\n",
      "If you use programs that use GNU Parallel to process data for an article in a\n",
      "scientific publication, please cite:\n",
      "\n",
      "  Tange, O. (2024, May 22). GNU Parallel 20240522 ('Tbilisi').\n",
      "  Zenodo. https://doi.org/10.5281/zenodo.11247979\n",
      "\n",
      "This helps funding further development; AND IT WON'T COST YOU A CENT.\n",
      "If you pay 10000 EUR you should feel free to use GNU Parallel without citing.\n",
      "\n",
      "More about funding GNU Parallel and the citation notice:\n",
      "https://www.gnu.org/software/parallel/parallel_design.html#citation-notice\n",
      "\n",
      "To silence this citation notice: run 'parallel --citation' once.\n",
      "\n",
      "Come on: You have run parallel 46 times. Isn't it about time \n",
      "you run 'parallel --citation' once to silence the citation notice?\n",
      "\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_SC102A1/Reproducibility_SC102A1_2_10\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_SC102A1/Reproducibility_SC102A1_1_10\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_SC102A1/Reproducibility_SC102A1_1_4\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_SC102A1/Reproducibility_SC102A1_1_3\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_SC102A1/Reproducibility_SC102A1_1_11\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_SC102A1/Reproducibility_SC102A1_1_5\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_SC102A1/Reproducibility_SC102A1_1_2\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_SC102A1/Reproducibility_SC102A1_1_8\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_SC102A1/Reproducibility_SC102A1_2_11\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_SC102A1/Reproducibility_SC102A1_1_6\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_SC102A1/Reproducibility_SC102A1_2_1\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_SC102A1/Reproducibility_SC102A1_1_12\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_SC102A1/Reproducibility_SC102A1_1_1\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_SC102A1/Reproducibility_SC102A1_1_9\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_SC102A1/Reproducibility_SC102A1_1_7\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_SC102A1/Reproducibility_SC102A1_2_12\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_SC102A1/Reproducibility_SC102A1_2_6\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_SC102A1/Reproducibility_SC102A1_2_8\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_SC102A1/Reproducibility_SC102A1_2_5\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_SC102A1/Reproducibility_SC102A1_2_9\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_SC102A1/Reproducibility_SC102A1_2_2\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_SC102A1/Reproducibility_SC102A1_2_7\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_SC102A1/Reproducibility_SC102A1_2_4\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_SC102A1/Reproducibility_SC102A1_2_3\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_SC102A1/Reproducibility_SC102A1_3_1\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_SC102A1/Reproducibility_SC102A1_3_3\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_SC102A1/Reproducibility_SC102A1_3_4\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_SC102A1/Reproducibility_SC102A1_3_10\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_SC102A1/Reproducibility_SC102A1_3_11\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_SC102A1/Reproducibility_SC102A1_3_2\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_SC102A1/Reproducibility_SC102A1_3_12\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_SC102A1/Reproducibility_SC102A1_3_9\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_SC102A1/Reproducibility_SC102A1_3_5\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_SC102A1/Reproducibility_SC102A1_3_6\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_SC102A1/Reproducibility_SC102A1_3_8\n",
      "/nas/groups/treutlein/USERS/jjans/analysis/iNeuron_morphogens/revisions/bulk_experiments/Reproducibility_SC102A1/Reproducibility_SC102A1_3_7\n",
      "(bulk_seq) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "export MAXCORES=8\n",
    "export STARGEN=/links/groups/treutlein/USERS/jjans/resources/genomes/STAR/Homo_sapiens.GRCh38\n",
    "\n",
    "ls /links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Reproducibility_SC102A1*/*S1*R1*.fastq.gz | parallel -j 15 --plus --rpl '{/..L} s:.fastq.gz::;s:.*/::;' --rpl '{/..} s:.fastq.gz::;s:.*/::;s:_S[0-9].*::' \"\n",
    "        cd Reproducibility_SC102A1/{/..}\n",
    "        \n",
    "        mkdir -p 10.MAPPING\n",
    "        \n",
    "        STAR \\\n",
    "        --runThreadN ${MAXCORES} \\\n",
    "        --genomeDir ${STARGEN} \\\n",
    "        --readFilesIn 00.CLEAN_FQ/*merged*.fastq.gz \\\n",
    "        --readFilesCommand zcat \\\n",
    "        --outFileNamePrefix 10.MAPPING/{/..}_STAR_ \\\n",
    "        --outSAMtype BAM SortedByCoordinate \\\n",
    "        --outWigType bedGraph \\\n",
    "        --outWigStrand Unstranded \\\n",
    "        --outWigNorm RPM \\\n",
    "        --outReadsUnmapped Fastx \\\n",
    "        --quantMode GeneCounts \\\n",
    "        &>> {/..}_STAR.log\n",
    "\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "07a8533d-a56c-49cd-b7fc-47c02e81caa2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(bulk_seq) \n",
      "(bulk_seq) \n",
      "(bulk_seq) \n",
      "Academic tradition requires you to cite works you base your article on.\n",
      "If you use programs that use GNU Parallel to process data for an article in a\n",
      "scientific publication, please cite:\n",
      "\n",
      "  Tange, O. (2024, May 22). GNU Parallel 20240522 ('Tbilisi').\n",
      "  Zenodo. https://doi.org/10.5281/zenodo.11247979\n",
      "\n",
      "This helps funding further development; AND IT WON'T COST YOU A CENT.\n",
      "If you pay 10000 EUR you should feel free to use GNU Parallel without citing.\n",
      "\n",
      "More about funding GNU Parallel and the citation notice:\n",
      "https://www.gnu.org/software/parallel/parallel_design.html#citation-notice\n",
      "\n",
      "To silence this citation notice: run 'parallel --citation' once.\n",
      "\n",
      "Come on: You have run parallel 45 times. Isn't it about time \n",
      "you run 'parallel --citation' once to silence the citation notice?\n",
      "\n",
      "Reproducibility_SC102A1_1_1\n",
      "/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Reproducibility_SC102A1_1_1/Reproducibility_SC102A1_1_1_S1_L001_R1_001.fastq.gz\n",
      "Reproducibility_SC102A1_1_1_S1_L001_R1_001\n",
      "Reproducibility_SC102A1_1_10\n",
      "/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Reproducibility_SC102A1_1_10/Reproducibility_SC102A1_1_10_S1_L001_R1_001.fastq.gz\n",
      "Reproducibility_SC102A1_1_10_S1_L001_R1_001\n",
      "Reproducibility_SC102A1_1_11\n",
      "/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Reproducibility_SC102A1_1_11/Reproducibility_SC102A1_1_11_S1_L001_R1_001.fastq.gz\n",
      "Reproducibility_SC102A1_1_11_S1_L001_R1_001\n",
      "Reproducibility_SC102A1_1_12\n",
      "/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Reproducibility_SC102A1_1_12/Reproducibility_SC102A1_1_12_S1_L001_R1_001.fastq.gz\n",
      "Reproducibility_SC102A1_1_12_S1_L001_R1_001\n",
      "Reproducibility_SC102A1_1_2\n",
      "/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Reproducibility_SC102A1_1_2/Reproducibility_SC102A1_1_2_S1_L001_R1_001.fastq.gz\n",
      "Reproducibility_SC102A1_1_2_S1_L001_R1_001\n",
      "Reproducibility_SC102A1_1_3\n",
      "/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Reproducibility_SC102A1_1_3/Reproducibility_SC102A1_1_3_S1_L002_R1_001.fastq.gz\n",
      "Reproducibility_SC102A1_1_3_S1_L002_R1_001\n",
      "Reproducibility_SC102A1_1_4\n",
      "/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Reproducibility_SC102A1_1_4/Reproducibility_SC102A1_1_4_S1_L002_R1_001.fastq.gz\n",
      "Reproducibility_SC102A1_1_4_S1_L002_R1_001\n",
      "Reproducibility_SC102A1_1_5\n",
      "/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Reproducibility_SC102A1_1_5/Reproducibility_SC102A1_1_5_S1_L002_R1_001.fastq.gz\n",
      "Reproducibility_SC102A1_1_5_S1_L002_R1_001\n",
      "Reproducibility_SC102A1_1_6\n",
      "/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Reproducibility_SC102A1_1_6/Reproducibility_SC102A1_1_6_S1_L001_R1_001.fastq.gz\n",
      "Reproducibility_SC102A1_1_6_S1_L001_R1_001\n",
      "Reproducibility_SC102A1_1_7\n",
      "/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Reproducibility_SC102A1_1_7/Reproducibility_SC102A1_1_7_S1_L001_R1_001.fastq.gz\n",
      "Reproducibility_SC102A1_1_7_S1_L001_R1_001\n",
      "Reproducibility_SC102A1_1_8\n",
      "/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Reproducibility_SC102A1_1_8/Reproducibility_SC102A1_1_8_S1_L002_R1_001.fastq.gz\n",
      "Reproducibility_SC102A1_1_8_S1_L002_R1_001\n",
      "Reproducibility_SC102A1_1_9\n",
      "/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Reproducibility_SC102A1_1_9/Reproducibility_SC102A1_1_9_S1_L002_R1_001.fastq.gz\n",
      "Reproducibility_SC102A1_1_9_S1_L002_R1_001\n",
      "Reproducibility_SC102A1_2_1\n",
      "/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Reproducibility_SC102A1_2_1/Reproducibility_SC102A1_2_1_S1_L001_R1_001.fastq.gz\n",
      "Reproducibility_SC102A1_2_1_S1_L001_R1_001\n",
      "Reproducibility_SC102A1_2_10\n",
      "/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Reproducibility_SC102A1_2_10/Reproducibility_SC102A1_2_10_S1_L001_R1_001.fastq.gz\n",
      "Reproducibility_SC102A1_2_10_S1_L001_R1_001\n",
      "Reproducibility_SC102A1_2_11\n",
      "/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Reproducibility_SC102A1_2_11/Reproducibility_SC102A1_2_11_S1_L002_R1_001.fastq.gz\n",
      "Reproducibility_SC102A1_2_11_S1_L002_R1_001\n",
      "Reproducibility_SC102A1_2_12\n",
      "/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Reproducibility_SC102A1_2_12/Reproducibility_SC102A1_2_12_S1_L002_R1_001.fastq.gz\n",
      "Reproducibility_SC102A1_2_12_S1_L002_R1_001\n",
      "Reproducibility_SC102A1_2_2\n",
      "/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Reproducibility_SC102A1_2_2/Reproducibility_SC102A1_2_2_S1_L002_R1_001.fastq.gz\n",
      "Reproducibility_SC102A1_2_2_S1_L002_R1_001\n",
      "Reproducibility_SC102A1_2_3\n",
      "/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Reproducibility_SC102A1_2_3/Reproducibility_SC102A1_2_3_S1_L001_R1_001.fastq.gz\n",
      "Reproducibility_SC102A1_2_3_S1_L001_R1_001\n",
      "Reproducibility_SC102A1_2_4\n",
      "/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Reproducibility_SC102A1_2_4/Reproducibility_SC102A1_2_4_S1_L001_R1_001.fastq.gz\n",
      "Reproducibility_SC102A1_2_4_S1_L001_R1_001\n",
      "Reproducibility_SC102A1_2_5\n",
      "/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Reproducibility_SC102A1_2_5/Reproducibility_SC102A1_2_5_S1_L002_R1_001.fastq.gz\n",
      "Reproducibility_SC102A1_2_5_S1_L002_R1_001\n",
      "Reproducibility_SC102A1_2_6\n",
      "/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Reproducibility_SC102A1_2_6/Reproducibility_SC102A1_2_6_S1_L002_R1_001.fastq.gz\n",
      "Reproducibility_SC102A1_2_6_S1_L002_R1_001\n",
      "Reproducibility_SC102A1_2_7\n",
      "/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Reproducibility_SC102A1_2_7/Reproducibility_SC102A1_2_7_S1_L002_R1_001.fastq.gz\n",
      "Reproducibility_SC102A1_2_7_S1_L002_R1_001\n",
      "Reproducibility_SC102A1_2_8\n",
      "/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Reproducibility_SC102A1_2_8/Reproducibility_SC102A1_2_8_S1_L002_R1_001.fastq.gz\n",
      "Reproducibility_SC102A1_2_8_S1_L002_R1_001\n",
      "Reproducibility_SC102A1_2_9\n",
      "/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Reproducibility_SC102A1_2_9/Reproducibility_SC102A1_2_9_S1_L001_R1_001.fastq.gz\n",
      "Reproducibility_SC102A1_2_9_S1_L001_R1_001\n",
      "Reproducibility_SC102A1_3_1\n",
      "/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Reproducibility_SC102A1_3_1/Reproducibility_SC102A1_3_1_S1_L002_R1_001.fastq.gz\n",
      "Reproducibility_SC102A1_3_1_S1_L002_R1_001\n",
      "Reproducibility_SC102A1_3_10\n",
      "/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Reproducibility_SC102A1_3_10/Reproducibility_SC102A1_3_10_S1_L002_R1_001.fastq.gz\n",
      "Reproducibility_SC102A1_3_10_S1_L002_R1_001\n",
      "Reproducibility_SC102A1_3_11\n",
      "/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Reproducibility_SC102A1_3_11/Reproducibility_SC102A1_3_11_S1_L001_R1_001.fastq.gz\n",
      "Reproducibility_SC102A1_3_11_S1_L001_R1_001\n",
      "Reproducibility_SC102A1_3_12\n",
      "/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Reproducibility_SC102A1_3_12/Reproducibility_SC102A1_3_12_S1_L001_R1_001.fastq.gz\n",
      "Reproducibility_SC102A1_3_12_S1_L001_R1_001\n",
      "Reproducibility_SC102A1_3_2\n",
      "/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Reproducibility_SC102A1_3_2/Reproducibility_SC102A1_3_2_S1_L002_R1_001.fastq.gz\n",
      "Reproducibility_SC102A1_3_2_S1_L002_R1_001\n",
      "Reproducibility_SC102A1_3_3\n",
      "/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Reproducibility_SC102A1_3_3/Reproducibility_SC102A1_3_3_S1_L002_R1_001.fastq.gz\n",
      "Reproducibility_SC102A1_3_3_S1_L002_R1_001\n",
      "Reproducibility_SC102A1_3_4\n",
      "/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Reproducibility_SC102A1_3_4/Reproducibility_SC102A1_3_4_S1_L001_R1_001.fastq.gz\n",
      "Reproducibility_SC102A1_3_4_S1_L001_R1_001\n",
      "Reproducibility_SC102A1_3_5\n",
      "/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Reproducibility_SC102A1_3_5/Reproducibility_SC102A1_3_5_S1_L001_R1_001.fastq.gz\n",
      "Reproducibility_SC102A1_3_5_S1_L001_R1_001\n",
      "Reproducibility_SC102A1_3_6\n",
      "/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Reproducibility_SC102A1_3_6/Reproducibility_SC102A1_3_6_S1_L001_R1_001.fastq.gz\n",
      "Reproducibility_SC102A1_3_6_S1_L001_R1_001\n",
      "Reproducibility_SC102A1_3_7\n",
      "/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Reproducibility_SC102A1_3_7/Reproducibility_SC102A1_3_7_S1_L002_R1_001.fastq.gz\n",
      "Reproducibility_SC102A1_3_7_S1_L002_R1_001\n",
      "Reproducibility_SC102A1_3_8\n",
      "/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Reproducibility_SC102A1_3_8/Reproducibility_SC102A1_3_8_S1_L001_R1_001.fastq.gz\n",
      "Reproducibility_SC102A1_3_8_S1_L001_R1_001\n",
      "Reproducibility_SC102A1_3_9\n",
      "/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Reproducibility_SC102A1_3_9/Reproducibility_SC102A1_3_9_S1_L001_R1_001.fastq.gz\n",
      "Reproducibility_SC102A1_3_9_S1_L001_R1_001\n",
      "(bulk_seq) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "export MAXCORES=8\n",
    "export STARGEN=/links/groups/treutlein/USERS/jjans/resources/genomes/STAR/Homo_sapiens.GRCh38\n",
    "\n",
    "ls /links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/Reproducibility_SC102A1*/*S1*R1*.fastq.gz | parallel -j 15 --plus --rpl '{/..L} s:.fastq.gz::;s:.*/::;' --rpl '{/..} s:.fastq.gz::;s:.*/::;s:_S[0-9].*::' \"\n",
    "echo {/..}\n",
    "echo {}\n",
    "echo {/..L}\n",
    "\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "981434e1-9702-488c-887d-66949ab4c3cf",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "5460595a-72ba-4067-95f8-0b01383abe09",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;32m00_FASTP_FQ_FILES.ipynb\u001b[0m                 \u001b[01;34mStability_d21\u001b[0m\n",
      "\u001b[01;32m01_bulk_analysis-iGABA_pre.ipynb\u001b[0m        \u001b[01;34mStability_d28\u001b[0m\n",
      "\u001b[01;32m01_bulk_analysis-iGlut_pre-Copy2.ipynb\u001b[0m  \u001b[01;34mStability_d35\u001b[0m\n",
      "\u001b[01;32m01_bulk_analysis-iGlut_pre.ipynb\u001b[0m        \u001b[01;34mStability_d42\u001b[0m\n",
      "\u001b[01;32m02_Stability_time.ipynb\u001b[0m                 \u001b[01;34mStability_d49\u001b[0m\n",
      "\u001b[01;32m02_bulk_analysis_stability_1.ipynb\u001b[0m      \u001b[01;32mUntitled.ipynb\u001b[0m\n",
      "\u001b[01;32m03_bulk_analysis_stability.ipynb\u001b[0m        \u001b[01;34mbulk\u001b[0m\n",
      "\u001b[01;32m10_Mapping_STAR.ipynb\u001b[0m                   \u001b[01;34miGABA_pre\u001b[0m\n",
      "\u001b[01;34mReproducibility_409B2\u001b[0m                   \u001b[01;32mprepattern_analysis_python.ipynb\u001b[0m\n",
      "\u001b[01;34mReproducibility_SC102A1\u001b[0m                 \u001b[01;34mstability_plots\u001b[0m\n",
      "\u001b[01;34mStability_d10\u001b[0m                           \u001b[01;32mstability_python.ipynb\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4012e888-ff29-4d02-b063-96aa2aa77cdd",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3fc2e546-cbce-4c91-8eeb-ec05bbe0e1ef",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "63df633b-b6f6-4165-8eeb-9f83efa81dfd",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4ae9d83f-f27e-4f32-ab63-0db4f2370d3a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "id": "db88a58f-f366-46fb-b9aa-a55ddf390b13",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(bulk_seq) \n",
      "(bulk_seq) \n",
      "(bulk_seq) \n",
      "(bulk_seq) \n",
      "Reproducibility_SC102A1_1_1\n",
      "Reproducibility_SC102A1_1_10\n",
      "Reproducibility_SC102A1_1_11\n",
      "Reproducibility_SC102A1_1_12\n",
      "Reproducibility_SC102A1_1_2\n",
      "Reproducibility_SC102A1_1_3\n",
      "Reproducibility_SC102A1_1_4\n",
      "Reproducibility_SC102A1_1_5\n",
      "Reproducibility_SC102A1_1_6\n",
      "Reproducibility_SC102A1_1_7\n",
      "Reproducibility_SC102A1_1_8\n",
      "Reproducibility_SC102A1_1_9\n",
      "Reproducibility_SC102A1_2_1\n",
      "Reproducibility_SC102A1_2_10\n",
      "Reproducibility_SC102A1_2_11\n",
      "Reproducibility_SC102A1_2_12\n",
      "Reproducibility_SC102A1_2_2\n",
      "Reproducibility_SC102A1_2_3\n",
      "Reproducibility_SC102A1_2_4\n",
      "Reproducibility_SC102A1_2_5\n",
      "Reproducibility_SC102A1_2_6\n",
      "Reproducibility_SC102A1_2_7\n",
      "Reproducibility_SC102A1_2_8\n",
      "Reproducibility_SC102A1_2_9\n",
      "Reproducibility_SC102A1_3_1\n",
      "Reproducibility_SC102A1_3_10\n",
      "Reproducibility_SC102A1_3_11\n",
      "Reproducibility_SC102A1_3_12\n",
      "Reproducibility_SC102A1_3_2\n",
      "Reproducibility_SC102A1_3_3\n",
      "Reproducibility_SC102A1_3_4\n",
      "Reproducibility_SC102A1_3_5\n",
      "Reproducibility_SC102A1_3_6\n",
      "Reproducibility_SC102A1_3_7\n",
      "Reproducibility_SC102A1_3_8\n",
      "Reproducibility_SC102A1_3_9\n",
      "(bulk_seq) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "# Define the base directory\n",
    "BASE_DIR=\"Reproducibility_SC102A1\"\n",
    "\n",
    "# Merge FASTQ files within each folder\n",
    "for FOLDER in $BASE_DIR/*; do\n",
    "    # Skip if not a directory\n",
    "    if [ ! -d \"$FOLDER\" ]; then\n",
    "        continue\n",
    "    fi\n",
    "    \n",
    "    # Get the folder name\n",
    "    FOLDER_NAME=$(basename \"$FOLDER\")\n",
    "    \n",
    "    echo $FOLDER_NAME\n",
    "    \n",
    "    mkdir -p 10.MAPPING\n",
    "\n",
    "    STAR \\\n",
    "    --runThreadN ${MAXCORES} \\\n",
    "    --genomeDir ${STARGEN} \\\n",
    "    --readFilesIn 00.CLEAN_FQ/*merged*.fastq.gz \\\n",
    "    --readFilesCommand zcat \\\n",
    "    --outFileNamePrefix 10.MAPPING/{/..}_STAR_ \\\n",
    "    --outSAMtype BAM SortedByCoordinate \\\n",
    "    --outWigType bedGraph \\\n",
    "    --outWigStrand Unstranded \\\n",
    "    --outWigNorm RPM \\\n",
    "    --outReadsUnmapped Fastx \\\n",
    "    --quantMode GeneCounts \\\n",
    "    &>> STAR.log\n",
    "\n",
    "done\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cb342187-b43f-42fa-9c3d-121725ba3df3",
   "metadata": {},
   "outputs": [],
   "source": [
    "    # Merge FASTQ files within the folder\n",
    "    cd \"$FOLDER\"/00.CLEAN_FQ\n",
    "    cat *_R1.clean.fastq.gz > \"$FOLDER_NAME\"_R1.merged.fastq.gz\n",
    "    cat *_R2.clean.fastq.gz > \"$FOLDER_NAME\"_R2.merged.fastq.gz\n",
    "    cd ../../..\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "63e91833-e84f-4e3d-b220-f729fba81be1",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "id": "c1ecb7ad-7f25-4158-bea7-d8c13fb17034",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;32m00_FASTP_FQ_FILES.ipynb\u001b[0m  \u001b[01;34mStability_d10\u001b[0m  \u001b[01;34mStability_d35\u001b[0m  \u001b[01;32mUntitled.ipynb\u001b[0m\n",
      "\u001b[01;34mReproducibility_409B2\u001b[0m    \u001b[01;34mStability_d21\u001b[0m  \u001b[01;34mStability_d42\u001b[0m  \u001b[01;32mUntitled1.ipynb\u001b[0m\n",
      "\u001b[01;34mReproducibility_SC102A1\u001b[0m  \u001b[01;34mStability_d28\u001b[0m  \u001b[01;34mStability_d49\u001b[0m  \u001b[01;34miGABA_pre\u001b[0m\n",
      "(bulk_seq) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e38b20f7-cac6-416a-b074-cfcf193f621b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "97d85eec-ae10-4a82-9a8d-4cc046a15d1f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "83b5e2a9-1693-4113-9ff1-e2073f548be9",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "66853e97-b6fe-47e9-8c11-a9d81669b5d2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;32m00_FASTP_FQ_FILES.ipynb\u001b[0m  \u001b[01;34mStability_d10\u001b[0m  \u001b[01;34mStability_d35\u001b[0m  \u001b[01;32mUntitled.ipynb\u001b[0m\n",
      "\u001b[01;34mReproducibility_409B2\u001b[0m    \u001b[01;34mStability_d21\u001b[0m  \u001b[01;34mStability_d42\u001b[0m  \u001b[01;32mUntitled1.ipynb\u001b[0m\n",
      "\u001b[01;34mReproducibility_SC102A1\u001b[0m  \u001b[01;34mStability_d28\u001b[0m  \u001b[01;34mStability_d49\u001b[0m  \u001b[01;34miGABA_pre\u001b[0m\n",
      "(bulk_seq) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8fc0481d-bb99-4afe-a222-2bcc8cabe70e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1af106a7-b4d0-4fe4-9163-5b3833129169",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a14c3d6a-25a6-4d8d-a786-6b259fa880cb",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c4d397ef-0f6d-4b93-844b-479e9b794364",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8ff38bd8-2902-412b-b826-64aee3931953",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ef94a281-19ce-4591-a657-b12017a6eb7c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9694a90e-be64-47ca-9bb0-30015750d734",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dfb212d4-d4da-44c3-b0a6-2e58beabd7fa",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e849f207-12bf-446e-8172-024d6665a68e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8955a39a-3ecc-481b-b8d4-162dc0bd0615",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c154a87b-109c-4e22-8426-f3a0c5c926c7",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a67ac652-a3aa-42f7-ac55-3428c3e9c0e1",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c6878a7b-f44e-44f9-8a12-8367ac1bc003",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "c78398ca-5f57-4cbf-946d-0330a1a23cc1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;32m/links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_24/iGABA_pre_24_S1_L001_R1_001.fastq.gz\u001b[0m\u001b[K\n",
      "(bulk_seq) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "ls /links/groups/treutlein/DATA/sequencing/20240524_P2794_HSIU-CHUAN/raw/iGABA_pre_24/*S1*R1*.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "486a1e2f-a9cf-43fe-8657-9e351b2e0f1d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4f021ccb-06a2-4e9f-a8e6-5da5be86f483",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "31d8671c-a010-451a-825a-069fd76165e4",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "fe87578f-d8e1-43c6-a283-210ddb9531f2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Genome         chrNameLength.txt  genomeParameters.txt\n",
      "SA             chrStart.txt       sjdbInfo.txt\n",
      "SAindex        exonGeTrInfo.tab   sjdbList.fromGTF.out.tab\n",
      "chrLength.txt  exonInfo.tab       sjdbList.out.tab\n",
      "chrName.txt    geneInfo.tab       transcriptInfo.tab\n",
      "(bulk_seq) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "ls "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "8429a995-68b1-48be-8b33-3a1cb8a708d0",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Usage: STAR  [options]... --genomeDir /path/to/genome/index/   --readFilesIn R1.fq R2.fq\n",
      "Spliced Transcripts Alignment to a Reference (c) Alexander Dobin, 2009-2022\n",
      "\n",
      "STAR version=2.7.11b\n",
      "STAR compilation time,server,dir=2024-03-19T08:38:59+0000 :/opt/conda/conda-bld/star_1710837244939/work/source\n",
      "For more details see:\n",
      "<https://github.com/alexdobin/STAR>\n",
      "<https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf>\n",
      "### versions\n",
      "versionGenome           2.7.4a\n",
      "    string: earliest genome index version compatible with this STAR release. Please do not change this value!\n",
      "\n",
      "### Parameter Files\n",
      "parametersFiles          -\n",
      "    string: name of a user-defined parameters file, \"-\": none. Can only be defined on the command line.\n",
      "\n",
      "### System\n",
      "sysShell            -\n",
      "    string: path to the shell binary, preferably bash, e.g. /bin/bash.\n",
      "                    - ... the default shell is executed, typically /bin/sh. This was reported to fail on some Ubuntu systems - then you need to specify path to bash.\n",
      "\n",
      "### Run Parameters\n",
      "runMode                         alignReads\n",
      "    string: type of the run.\n",
      "                                alignReads             ... map reads\n",
      "                                genomeGenerate         ... generate genome files\n",
      "                                inputAlignmentsFromBAM ... input alignments from BAM. Presently only works with --outWigType and --bamRemoveDuplicates options.\n",
      "                                liftOver               ... lift-over of GTF files (--sjdbGTFfile) between genome assemblies using chain file(s) from --genomeChainFiles.\n",
      "                                soloCellFiltering  </path/to/raw/count/dir/>   </path/to/output/prefix>    ... STARsolo cell filtering (\"calling\") without remapping, followed by the path to raw count directory and output (filtered) prefix\n",
      "\n",
      "runThreadN                      1\n",
      "    int: number of threads to run STAR\n",
      "\n",
      "runDirPerm                      User_RWX\n",
      "    string: permissions for the directories created at the run-time.\n",
      "                                User_RWX ... user-read/write/execute\n",
      "                                All_RWX  ... all-read/write/execute (same as chmod 777)\n",
      "\n",
      "runRNGseed                      777\n",
      "    int: random number generator seed.\n",
      "\n",
      "\n",
      "### Genome Parameters\n",
      "genomeDir                   ./GenomeDir/\n",
      "    string: path to the directory where genome files are stored (for --runMode alignReads) or will be generated (for --runMode generateGenome)\n",
      "\n",
      "genomeLoad                NoSharedMemory\n",
      "    string: mode of shared memory usage for the genome files. Only used with --runMode alignReads.\n",
      "                          LoadAndKeep     ... load genome into shared and keep it in memory after run\n",
      "                          LoadAndRemove   ... load genome into shared but remove it after run\n",
      "                          LoadAndExit     ... load genome into shared memory and exit, keeping the genome in memory for future runs\n",
      "                          Remove          ... do not map anything, just remove loaded genome from memory\n",
      "                          NoSharedMemory  ... do not use shared memory, each job will have its own private copy of the genome\n",
      "\n",
      "genomeFastaFiles            -\n",
      "    string(s): path(s) to the fasta files with the genome sequences, separated by spaces. These files should be plain text FASTA files, they *cannot* be zipped.\n",
      "                            Required for the genome generation (--runMode genomeGenerate). Can also be used in the mapping (--runMode alignReads) to add extra (new) sequences to the genome (e.g. spike-ins).\n",
      "\n",
      "genomeChainFiles            -\n",
      "    string: chain files for genomic liftover. Only used with --runMode liftOver .\n",
      "\n",
      "genomeFileSizes             0\n",
      "    uint(s)>0: genome files exact sizes in bytes. Typically, this should not be defined by the user.\n",
      "    \n",
      "genomeTransformOutput       None\n",
      "    string(s):              which output to transform back to original genome\n",
      "                            SAM     ... SAM/BAM alignments\n",
      "                            SJ      ... splice junctions (SJ.out.tab)\n",
      "                            Quant   ... quantifications (from --quantMode option)\n",
      "                            None    ... no transformation of the output        \n",
      "\n",
      "genomeChrSetMitochondrial   chrM M MT\n",
      "    string(s):              names of the mitochondrial chromosomes. Presently only used for STARsolo statistics output/\n",
      "\n",
      "### Genome Indexing Parameters - only used with --runMode genomeGenerate\n",
      "genomeChrBinNbits           18\n",
      "    int: =log2(chrBin), where chrBin is the size of the bins for genome storage: each chromosome will occupy an integer number of bins. For a genome with large number of contigs, it is recommended to scale this parameter as min(18, log2[max(GenomeLength/NumberOfReferences,ReadLength)]).\n",
      "\n",
      "genomeSAindexNbases         14\n",
      "    int: length (bases) of the SA pre-indexing string. Typically between 10 and 15. Longer strings will use much more memory, but allow faster searches. For small genomes, the parameter --genomeSAindexNbases must be scaled down to min(14, log2(GenomeLength)/2 - 1).\n",
      "\n",
      "genomeSAsparseD             1\n",
      "    int>0: suffux array sparsity, i.e. distance between indices: use bigger numbers to decrease needed RAM at the cost of mapping speed reduction\n",
      "\n",
      "genomeSuffixLengthMax       -1\n",
      "    int: maximum length of the suffixes, has to be longer than read length. -1 = infinite.\n",
      "    \n",
      "genomeTransformType         None\n",
      "    string: type of genome transformation\n",
      "                            None       ... no transformation\n",
      "                            Haploid    ... replace reference alleles with alternative alleles from VCF file (e.g. consensus allele)\n",
      "                            Diploid    ... create two haplotypes for each chromosome listed in VCF file, for genotypes 1|2, assumes perfect phasing (e.g. personal genome)\n",
      "\n",
      "genomeTransformVCF          -\n",
      "    string: path to VCF file for genome transformation\n",
      "\n",
      "\n",
      "    \n",
      "#####UnderDevelopment_begin : not supported - do not use\n",
      "genomeType                  Full\n",
      "    string: type of genome to generate\n",
      "                            Full                ... full (normal) genome\n",
      "                            Transcriptome       ... genome consists of transcript sequences\n",
      "                            SuperTransriptome   ... genome consists of superTranscript sequences\n",
      "#####UnderDevelopment_end\n",
      "\n",
      "# DEPRECATED: please use --genomeTransformVCF and --genomeTransformType options instead.\n",
      "#genomeConsensusFile         -\n",
      "#    string: VCF file with consensus SNPs (i.e. alternative allele is the major (AF>0.5) allele)\n",
      "# DEPRECATED \n",
      "\n",
      "\n",
      "\n",
      "### Splice Junctions Database\n",
      "sjdbFileChrStartEnd                     -\n",
      "    string(s): path to the files with genomic coordinates (chr <tab> start <tab> end <tab> strand) for the splice junction introns. Multiple files can be supplied and will be concatenated.\n",
      "\n",
      "sjdbGTFfile                             -\n",
      "    string: path to the GTF file with annotations\n",
      "\n",
      "sjdbGTFchrPrefix                        -\n",
      "    string: prefix for chromosome names in a GTF file (e.g. 'chr' for using ENSMEBL annotations with UCSC genomes)\n",
      "\n",
      "sjdbGTFfeatureExon                      exon\n",
      "    string: feature type in GTF file to be used as exons for building transcripts\n",
      "\n",
      "sjdbGTFtagExonParentTranscript          transcript_id\n",
      "    string: GTF attribute name for parent transcript ID (default \"transcript_id\" works for GTF files)\n",
      "\n",
      "sjdbGTFtagExonParentGene                gene_id\n",
      "    string: GTF attribute name for parent gene ID (default \"gene_id\" works for GTF files)\n",
      "\n",
      "sjdbGTFtagExonParentGeneName            gene_name\n",
      "    string(s): GTF attribute name for parent gene name\n",
      "\n",
      "sjdbGTFtagExonParentGeneType            gene_type gene_biotype\n",
      "    string(s): GTF attribute name for parent gene type\n",
      "\n",
      "sjdbOverhang                            100\n",
      "    int>0: length of the donor/acceptor sequence on each side of the junctions, ideally = (mate_length - 1)\n",
      "\n",
      "sjdbScore                               2\n",
      "    int: extra alignment score for alignments that cross database junctions\n",
      "\n",
      "sjdbInsertSave                          Basic\n",
      "    string: which files to save when sjdb junctions are inserted on the fly at the mapping step\n",
      "                    Basic ... only small junction / transcript files\n",
      "                    All   ... all files including big Genome, SA and SAindex - this will create a complete genome directory\n",
      "\n",
      "### Variation parameters\n",
      "varVCFfile                              -\n",
      "    string: path to the VCF file that contains variation data. The 10th column should contain the genotype information, e.g. 0/1\n",
      "\n",
      "### Input Files\n",
      "inputBAMfile                -\n",
      "    string: path to BAM input file, to be used with --runMode inputAlignmentsFromBAM\n",
      "\n",
      "### Read Parameters\n",
      "readFilesType               Fastx\n",
      "    string: format of input read files\n",
      "                            Fastx       ... FASTA or FASTQ\n",
      "                            SAM SE      ... SAM or BAM single-end reads; for BAM use --readFilesCommand samtools view\n",
      "                            SAM PE      ... SAM or BAM paired-end reads; for BAM use --readFilesCommand samtools view\n",
      "                            \n",
      "readFilesSAMattrKeep        All\n",
      "    string(s): for --readFilesType SAM SE/PE, which SAM tags to keep in the output BAM, e.g.: --readFilesSAMtagsKeep RG PL\n",
      "                            All     ... keep all tags\n",
      "                            None    ... do not keep any tags\n",
      "\n",
      "readFilesIn                 Read1 Read2\n",
      "    string(s): paths to files that contain input read1 (and, if needed,  read2)\n",
      "\n",
      "readFilesManifest           -\n",
      "    string: path to the \"manifest\" file with the names of read files. The manifest file should contain 3 tab-separated columns:\n",
      "            paired-end reads: read1_file_name $tab$ read2_file_name $tab$ read_group_line.\n",
      "            single-end reads: read1_file_name $tab$ -               $tab$ read_group_line.\n",
      "            Spaces, but not tabs are allowed in file names.\n",
      "            If read_group_line does not start with ID:, it can only contain one ID field, and ID: will be added to it.\n",
      "            If read_group_line starts with ID:, it can contain several fields separated by $tab$, and all fields will be be copied verbatim into SAM @RG header line.\n",
      "\n",
      "readFilesPrefix             -\n",
      "    string: prefix for the read files names, i.e. it will be added in front of the strings in --readFilesIn\n",
      "\n",
      "readFilesCommand             -\n",
      "    string(s): command line to execute for each of the input file. This command should generate FASTA or FASTQ text and send it to stdout\n",
      "               For example: zcat - to uncompress .gz files, bzcat - to uncompress .bz2 files, etc.\n",
      "\n",
      "readMapNumber               -1\n",
      "    int: number of reads to map from the beginning of the file\n",
      "                            -1: map all reads\n",
      "\n",
      "readMatesLengthsIn          NotEqual\n",
      "    string: Equal/NotEqual - lengths of names,sequences,qualities for both mates are the same  / not the same. NotEqual is safe in all situations.\n",
      "\n",
      "readNameSeparator           /\n",
      "    string(s): character(s) separating the part of the read names that will be trimmed in output (read name after space is always trimmed)\n",
      "\n",
      "readQualityScoreBase        33\n",
      "    int>=0: number to be subtracted from the ASCII code to get Phred quality score\n",
      "\n",
      "### Read Clipping\n",
      "\n",
      "clipAdapterType             Hamming\n",
      "    string:                 adapter clipping type\n",
      "                            Hamming ... adapter clipping based on Hamming distance, with the number of mismatches controlled by --clip5pAdapterMMp\n",
      "                            CellRanger4 ... 5p and 3p adapter clipping similar to CellRanger4. Utilizes Opal package by Martin Šošić: https://github.com/Martinsos/opal\n",
      "                            None ... no adapter clipping, all other clip* parameters are disregarded\n",
      "                            \n",
      "clip3pNbases                 0\n",
      "    int(s): number(s) of bases to clip from 3p of each mate. If one value is given, it will be assumed the same for both mates.\n",
      "\n",
      "clip3pAdapterSeq            -\n",
      "    string(s): adapter sequences to clip from 3p of each mate.  If one value is given, it will be assumed the same for both mates.\n",
      "                            polyA ... polyA sequence with the length equal to read length\n",
      "\n",
      "clip3pAdapterMMp            0.1\n",
      "    double(s): max proportion of mismatches for 3p adapter clipping for each mate.  If one value is given, it will be assumed the same for both mates.\n",
      "\n",
      "clip3pAfterAdapterNbases    0\n",
      "    int(s): number of bases to clip from 3p of each mate after the adapter clipping. If one value is given, it will be assumed the same for both mates.\n",
      "\n",
      "clip5pNbases                 0\n",
      "    int(s): number(s) of bases to clip from 5p of each mate. If one value is given, it will be assumed the same for both mates.\n",
      "\n",
      "#####UnderDevelopment_begin : not supported - do not use   \n",
      "clip5pAdapterSeq            -\n",
      "    string(s): adapter sequences to clip from 5p of each mate, separated by space.\n",
      "\n",
      "clip5pAdapterMMp            0.1\n",
      "    double(s): max proportion of mismatches for 5p adapter clipping for each mate, separated by space\n",
      "\n",
      "clip5pAfterAdapterNbases    0\n",
      "    int(s): number of bases to clip from 5p of each mate after the adapter clipping, separated by space.\n",
      "#####UnderDevelopment_end\n",
      "\n",
      "### Limits\n",
      "limitGenomeGenerateRAM               31000000000\n",
      "    int>0: maximum available RAM (bytes) for genome generation\n",
      "\n",
      "limitIObufferSize                    30000000 50000000\n",
      "    int(s)>0: max available buffers size (bytes) for input/output, per thread\n",
      "\n",
      "limitOutSAMoneReadBytes              100000\n",
      "    int>0: max size of the SAM record (bytes) for one read. Recommended value: >(2*(LengthMate1+LengthMate2+100)*outFilterMultimapNmax\n",
      "\n",
      "limitOutSJoneRead                    1000\n",
      "    int>0: max number of junctions for one read (including all multi-mappers)\n",
      "\n",
      "limitOutSJcollapsed                  1000000\n",
      "    int>0: max number of collapsed junctions\n",
      "\n",
      "limitBAMsortRAM                         0\n",
      "    int>=0: maximum available RAM (bytes) for sorting BAM. If =0, it will be set to the genome index size. 0 value can only be used with --genomeLoad NoSharedMemory option.\n",
      "\n",
      "limitSjdbInsertNsj                     1000000\n",
      "    int>=0: maximum number of junctions to be inserted to the genome on the fly at the mapping stage, including those from annotations and those detected in the 1st step of the 2-pass run\n",
      "\n",
      "limitNreadsSoft                        -1\n",
      "    int: soft limit on the number of reads\n",
      "\n",
      "### Output: general\n",
      "outFileNamePrefix               ./\n",
      "    string: output files name prefix (including full or relative path). Can only be defined on the command line.\n",
      "\n",
      "outTmpDir                       -\n",
      "    string: path to a directory that will be used as temporary by STAR. All contents of this directory will be removed!\n",
      "                                - ... the temp directory will default to outFileNamePrefix_STARtmp\n",
      "\n",
      "outTmpKeep                      None\n",
      "    string: whether to keep the temporary files after STAR runs is finished\n",
      "                                None ... remove all temporary files\n",
      "                                All ... keep all files\n",
      "\n",
      "outStd                          Log\n",
      "    string: which output will be directed to stdout (standard out)\n",
      "                                Log                    ... log messages\n",
      "                                SAM                    ... alignments in SAM format (which normally are output to Aligned.out.sam file), normal standard output will go into Log.std.out\n",
      "                                BAM_Unsorted           ... alignments in BAM format, unsorted. Requires --outSAMtype BAM Unsorted\n",
      "                                BAM_SortedByCoordinate ... alignments in BAM format, sorted by coordinate. Requires --outSAMtype BAM SortedByCoordinate\n",
      "                                BAM_Quant              ... alignments to transcriptome in BAM format, unsorted. Requires --quantMode TranscriptomeSAM\n",
      "\n",
      "outReadsUnmapped                None\n",
      "   string: output of unmapped and partially mapped (i.e. mapped only one mate of a paired end read) reads in separate file(s).\n",
      "                                None    ... no output\n",
      "                                Fastx   ... output in separate fasta/fastq files, Unmapped.out.mate1/2\n",
      "\n",
      "outQSconversionAdd              0\n",
      "   int: add this number to the quality score (e.g. to convert from Illumina to Sanger, use -31)\n",
      "\n",
      "outMultimapperOrder             Old_2.4\n",
      "    string: order of multimapping alignments in the output files\n",
      "                                Old_2.4             ... quasi-random order used before 2.5.0\n",
      "                                Random              ... random order of alignments for each multi-mapper. Read mates (pairs) are always adjacent, all alignment for each read stay together. This option will become default in the future releases.\n",
      "\n",
      "### Output: SAM and BAM\n",
      "outSAMtype                      SAM\n",
      "    strings: type of SAM/BAM output\n",
      "                                1st word:\n",
      "                                BAM  ... output BAM without sorting\n",
      "                                SAM  ... output SAM without sorting\n",
      "                                None ... no SAM/BAM output\n",
      "                                2nd, 3rd:\n",
      "                                Unsorted           ... standard unsorted\n",
      "                                SortedByCoordinate ... sorted by coordinate. This option will allocate extra memory for sorting which can be specified by --limitBAMsortRAM.\n",
      "\n",
      "outSAMmode                      Full\n",
      "    string: mode of SAM output\n",
      "                                None ... no SAM output\n",
      "                                Full ... full SAM output\n",
      "                                NoQS ... full SAM but without quality scores\n",
      "\n",
      "outSAMstrandField               None\n",
      "    string: Cufflinks-like strand field flag\n",
      "                                None        ... not used\n",
      "                                intronMotif ... strand derived from the intron motif. This option changes the output alignments: reads with inconsistent and/or non-canonical introns are filtered out.\n",
      "\n",
      "outSAMattributes                Standard\n",
      "    string(s): a string of desired SAM attributes, in the order desired for the output SAM. Tags can be listed in any combination/order.\n",
      "                                ***Presets:\n",
      "                                None        ... no attributes\n",
      "                                Standard    ... NH HI AS nM\n",
      "                                All         ... NH HI AS nM NM MD jM jI MC ch                                                                    \n",
      "                                ***Alignment:\n",
      "                                NH          ... number of loci the reads maps to: =1 for unique mappers, >1 for multimappers. Standard SAM tag.\n",
      "                                HI          ... multiple alignment index, starts with --outSAMattrIHstart (=1 by default). Standard SAM tag.\n",
      "                                AS          ... local alignment score, +1/-1 for matches/mismateches, score* penalties for indels and gaps. For PE reads, total score for two mates. Stadnard SAM tag.\n",
      "                                nM          ... number of mismatches. For PE reads, sum over two mates.\n",
      "                                NM          ... edit distance to the reference (number of mismatched + inserted + deleted bases) for each mate. Standard SAM tag.\n",
      "                                MD          ... string encoding mismatched and deleted reference bases (see standard SAM specifications). Standard SAM tag.\n",
      "                                jM          ... intron motifs for all junctions (i.e. N in CIGAR): 0: non-canonical; 1: GT/AG, 2: CT/AC, 3: GC/AG, 4: CT/GC, 5: AT/AC, 6: GT/AT. If splice junctions database is used, and a junction is annotated, 20 is added to its motif value.\n",
      "                                jI          ... start and end of introns for all junctions (1-based).\n",
      "                                XS          ... alignment strand according to --outSAMstrandField.\n",
      "                                MC          ... mate's CIGAR string. Standard SAM tag.\n",
      "                                ch          ... marks all segment of all chimeric alingments for --chimOutType WithinBAM output.\n",
      "                                cN          ... number of bases clipped from the read ends: 5' and 3'\n",
      "                                ***Variation:\n",
      "                                vA          ... variant allele\n",
      "                                vG          ... genomic coordinate of the variant overlapped by the read.\n",
      "                                vW          ... 1 - alignment passes WASP filtering; 2,3,4,5,6,7 - alignment does not pass WASP filtering. Requires --waspOutputMode SAMtag.\n",
      "                                ha          ... haplotype (1/2) when mapping to the diploid genome. Requires genome generated with --genomeTransformType Diploid .                               \n",
      "                                ***STARsolo:\n",
      "                                CR CY UR UY ... sequences and quality scores of cell barcodes and UMIs for the solo* demultiplexing.\n",
      "                                GX GN       ... gene ID and gene name for unique-gene reads.\n",
      "                                gx gn       ... gene IDs and gene names for unique- and multi-gene reads.\n",
      "                                CB UB       ... error-corrected cell barcodes and UMIs for solo* demultiplexing. Requires --outSAMtype BAM SortedByCoordinate.\n",
      "                                sM          ... assessment of CB and UMI.\n",
      "                                sS          ... sequence of the entire barcode (CB,UMI,adapter).\n",
      "                                sQ          ... quality of the entire barcode.\n",
      "                                sF          ... type of feature overlap and number of features for each alignment\n",
      "                                ***Unsupported/undocumented:\n",
      "                                rB          ... alignment block read/genomic coordinates.\n",
      "                                vR          ... read coordinate of the variant.\n",
      "\n",
      "outSAMattrIHstart               1\n",
      "    int>=0:                     start value for the IH attribute. 0 may be required by some downstream software, such as Cufflinks or StringTie.\n",
      "\n",
      "outSAMunmapped                  None\n",
      "    string(s): output of unmapped reads in the SAM format\n",
      "                                1st word:\n",
      "                                None   ... no output\n",
      "                                Within ... output unmapped reads within the main SAM file (i.e. Aligned.out.sam)\n",
      "                                2nd word:\n",
      "                                KeepPairs ... record unmapped mate for each alignment, and, in case of unsorted output, keep it adjacent to its mapped mate. Only affects multi-mapping reads.\n",
      "\n",
      "outSAMorder                     Paired\n",
      "    string: type of sorting for the SAM output\n",
      "                                Paired: one mate after the other for all paired alignments\n",
      "                                PairedKeepInputOrder: one mate after the other for all paired alignments, the order is kept the same as in the input FASTQ files\n",
      "\n",
      "outSAMprimaryFlag        OneBestScore\n",
      "    string: which alignments are considered primary - all others will be marked with 0x100 bit in the FLAG\n",
      "                                OneBestScore ... only one alignment with the best score is primary\n",
      "                                AllBestScore ... all alignments with the best score are primary\n",
      "\n",
      "outSAMreadID            Standard\n",
      "    string: read ID record type\n",
      "                                Standard ... first word (until space) from the FASTx read ID line, removing /1,/2 from the end\n",
      "                                Number   ... read number (index) in the FASTx file\n",
      "\n",
      "outSAMmapqUnique        255\n",
      "    int: 0 to 255: the MAPQ value for unique mappers\n",
      "\n",
      "outSAMflagOR           0\n",
      "    int: 0 to 65535: sam FLAG will be bitwise OR'd with this value, i.e. FLAG=FLAG | outSAMflagOR. This is applied after all flags have been set by STAR, and after outSAMflagAND. Can be used to set specific bits that are not set otherwise.\n",
      "\n",
      "outSAMflagAND           65535\n",
      "    int: 0 to 65535: sam FLAG will be bitwise AND'd with this value, i.e. FLAG=FLAG & outSAMflagOR. This is applied after all flags have been set by STAR, but before outSAMflagOR. Can be used to unset specific bits that are not set otherwise.\n",
      "\n",
      "outSAMattrRGline        -\n",
      "    string(s): SAM/BAM read group line. The first word contains the read group identifier and must start with \"ID:\", e.g. --outSAMattrRGline ID:xxx CN:yy \"DS:z z z\".\n",
      "            xxx will be added as RG tag to each output alignment. Any spaces in the tag values have to be double quoted.\n",
      "            Comma separated RG lines correspons to different (comma separated) input files in --readFilesIn. Commas have to be surrounded by spaces, e.g.\n",
      "            --outSAMattrRGline ID:xxx , ID:zzz \"DS:z z\" , ID:yyy DS:yyyy\n",
      "\n",
      "outSAMheaderHD          -\n",
      "    strings: @HD (header) line of the SAM header\n",
      "\n",
      "outSAMheaderPG          -\n",
      "    strings: extra @PG (software) line of the SAM header (in addition to STAR)\n",
      "\n",
      "outSAMheaderCommentFile -\n",
      "    string: path to the file with @CO (comment) lines of the SAM header\n",
      "\n",
      "outSAMfilter            None\n",
      "    string(s): filter the output into main SAM/BAM files\n",
      "                        KeepOnlyAddedReferences ... only keep the reads for which all alignments are to the extra reference sequences added with --genomeFastaFiles at the mapping stage.\n",
      "                        KeepAllAddedReferences ...  keep all alignments to the extra reference sequences added with --genomeFastaFiles at the mapping stage.\n",
      "\n",
      "\n",
      "outSAMmultNmax          -1\n",
      "    int: max number of multiple alignments for a read that will be output to the SAM/BAM files. Note that if this value is not equal to -1, the top scoring alignment will be output first\n",
      "                        -1 ... all alignments (up to --outFilterMultimapNmax) will be output\n",
      "\n",
      "outSAMtlen              1\n",
      "    int: calculation method for the TLEN field in the SAM/BAM files\n",
      "                        1 ... leftmost base of the (+)strand mate to rightmost base of the (-)mate. (+)sign for the (+)strand mate\n",
      "                        2 ... leftmost base of any mate to rightmost base of any mate. (+)sign for the mate with the leftmost base. This is different from 1 for overlapping mates with protruding ends\n",
      "\n",
      "outBAMcompression       1\n",
      "    int: -1 to 10  BAM compression level, -1=default compression (6?), 0=no compression, 10=maximum compression\n",
      "\n",
      "outBAMsortingThreadN    0\n",
      "    int: >=0: number of threads for BAM sorting. 0 will default to min(6,--runThreadN).\n",
      "\n",
      "outBAMsortingBinsN      50\n",
      "    int: >0:  number of genome bins for coordinate-sorting\n",
      "\n",
      "### BAM processing\n",
      "bamRemoveDuplicatesType  -\n",
      "    string: mark duplicates in the BAM file, for now only works with (i) sorted BAM fed with inputBAMfile, and (ii) for paired-end alignments only\n",
      "                        -                       ... no duplicate removal/marking\n",
      "                        UniqueIdentical         ... mark all multimappers, and duplicate unique mappers. The coordinates, FLAG, CIGAR must be identical\n",
      "                        UniqueIdenticalNotMulti  ... mark duplicate unique mappers but not multimappers.\n",
      "\n",
      "bamRemoveDuplicatesMate2basesN   0\n",
      "    int>0: number of bases from the 5' of mate 2 to use in collapsing (e.g. for RAMPAGE)\n",
      "\n",
      "### Output Wiggle\n",
      "outWigType          None\n",
      "    string(s): type of signal output, e.g. \"bedGraph\" OR \"bedGraph read1_5p\". Requires sorted BAM: --outSAMtype BAM SortedByCoordinate .\n",
      "                    1st word:\n",
      "                    None       ... no signal output\n",
      "                    bedGraph   ... bedGraph format\n",
      "                    wiggle     ... wiggle format\n",
      "                    2nd word:\n",
      "                    read1_5p   ... signal from only 5' of the 1st read, useful for CAGE/RAMPAGE etc\n",
      "                    read2      ... signal from only 2nd read\n",
      "\n",
      "outWigStrand        Stranded\n",
      "    string: strandedness of wiggle/bedGraph output\n",
      "                    Stranded   ...  separate strands, str1 and str2\n",
      "                    Unstranded ...  collapsed strands\n",
      "\n",
      "outWigReferencesPrefix    -\n",
      "    string: prefix matching reference names to include in the output wiggle file, e.g. \"chr\", default \"-\" - include all references\n",
      "\n",
      "outWigNorm              RPM\n",
      "    string: type of normalization for the signal\n",
      "                        RPM    ... reads per million of mapped reads\n",
      "                        None   ... no normalization, \"raw\" counts\n",
      "\n",
      "### Output Filtering\n",
      "outFilterType                   Normal\n",
      "    string: type of filtering\n",
      "                                Normal  ... standard filtering using only current alignment\n",
      "                                BySJout ... keep only those reads that contain junctions that passed filtering into SJ.out.tab\n",
      "\n",
      "outFilterMultimapScoreRange     1\n",
      "    int: the score range below the maximum score for multimapping alignments\n",
      "\n",
      "outFilterMultimapNmax           10\n",
      "    int: maximum number of loci the read is allowed to map to. Alignments (all of them) will be output only if the read maps to no more loci than this value.\n",
      "         Otherwise no alignments will be output, and the read will be counted as \"mapped to too many loci\" in the Log.final.out .\n",
      "\n",
      "outFilterMismatchNmax           10\n",
      "    int: alignment will be output only if it has no more mismatches than this value.\n",
      "\n",
      "outFilterMismatchNoverLmax      0.3\n",
      "    real: alignment will be output only if its ratio of mismatches to *mapped* length is less than or equal to this value.\n",
      "\n",
      "outFilterMismatchNoverReadLmax  1.0\n",
      "    real: alignment will be output only if its ratio of mismatches to *read* length is less than or equal to this value.\n",
      "\n",
      "\n",
      "outFilterScoreMin               0\n",
      "    int: alignment will be output only if its score is higher than or equal to this value.\n",
      "\n",
      "outFilterScoreMinOverLread      0.66\n",
      "    real: same as outFilterScoreMin, but normalized to read length (sum of mates' lengths for paired-end reads)\n",
      "\n",
      "outFilterMatchNmin              0\n",
      "    int: alignment will be output only if the number of matched bases is higher than or equal to this value.\n",
      "\n",
      "outFilterMatchNminOverLread     0.66\n",
      "    real: sam as outFilterMatchNmin, but normalized to the read length (sum of mates' lengths for paired-end reads).\n",
      "\n",
      "outFilterIntronMotifs           None\n",
      "    string: filter alignment using their motifs\n",
      "                None                           ... no filtering\n",
      "                RemoveNoncanonical             ... filter out alignments that contain non-canonical junctions\n",
      "                RemoveNoncanonicalUnannotated  ... filter out alignments that contain non-canonical unannotated junctions when using annotated splice junctions database. The annotated non-canonical junctions will be kept.\n",
      "\n",
      "outFilterIntronStrands          RemoveInconsistentStrands\n",
      "    string: filter alignments\n",
      "                RemoveInconsistentStrands      ... remove alignments that have junctions with inconsistent strands\n",
      "                None                           ... no filtering\n",
      "\n",
      "### Output splice junctions (SJ.out.tab)\n",
      "outSJtype                       Standard\n",
      "    string: type of splice junction output\n",
      "                                Standard    ... standard SJ.out.tab output\n",
      "                                None        ... no splice junction output\n",
      "\n",
      "### Output Filtering: Splice Junctions\n",
      "outSJfilterReads                All\n",
      "    string: which reads to consider for collapsed splice junctions output\n",
      "                                All     ... all reads, unique- and multi-mappers\n",
      "                                Unique  ... uniquely mapping reads only\n",
      "\n",
      "outSJfilterOverhangMin          30  12  12  12\n",
      "    4 integers:    minimum overhang length for splice junctions on both sides for: (1) non-canonical motifs, (2) GT/AG and CT/AC motif, (3) GC/AG and CT/GC motif, (4) AT/AC and GT/AT motif. -1 means no output for that motif\n",
      "                                does not apply to annotated junctions\n",
      "\n",
      "outSJfilterCountUniqueMin       3   1   1   1\n",
      "    4 integers: minimum uniquely mapping read count per junction for: (1) non-canonical motifs, (2) GT/AG and CT/AC motif, (3) GC/AG and CT/GC motif, (4) AT/AC and GT/AT motif. -1 means no output for that motif\n",
      "                                Junctions are output if one of outSJfilterCountUniqueMin OR outSJfilterCountTotalMin conditions are satisfied\n",
      "                                does not apply to annotated junctions\n",
      "\n",
      "outSJfilterCountTotalMin     3   1   1   1\n",
      "    4 integers: minimum total (multi-mapping+unique) read count per junction for: (1) non-canonical motifs, (2) GT/AG and CT/AC motif, (3) GC/AG and CT/GC motif, (4) AT/AC and GT/AT motif. -1 means no output for that motif\n",
      "                                Junctions are output if one of outSJfilterCountUniqueMin OR outSJfilterCountTotalMin conditions are satisfied\n",
      "                                does not apply to annotated junctions\n",
      "\n",
      "outSJfilterDistToOtherSJmin     10  0   5   10\n",
      "    4 integers>=0: minimum allowed distance to other junctions' donor/acceptor\n",
      "                                does not apply to annotated junctions\n",
      "\n",
      "outSJfilterIntronMaxVsReadN        50000 100000 200000\n",
      "    N integers>=0: maximum gap allowed for junctions supported by 1,2,3,,,N reads\n",
      "                                i.e. by default junctions supported by 1 read can have gaps <=50000b, by 2 reads: <=100000b, by 3 reads: <=200000. by >=4 reads any gap <=alignIntronMax\n",
      "                                does not apply to annotated junctions\n",
      "\n",
      "### Scoring\n",
      "scoreGap                     0\n",
      "    int: splice junction penalty (independent on intron motif)\n",
      "\n",
      "scoreGapNoncan               -8\n",
      "    int: non-canonical junction penalty (in addition to scoreGap)\n",
      "\n",
      "scoreGapGCAG                 -4\n",
      "    int: GC/AG and CT/GC junction penalty (in addition to scoreGap)\n",
      "\n",
      "scoreGapATAC                 -8\n",
      "    int: AT/AC  and GT/AT junction penalty  (in addition to scoreGap)\n",
      "\n",
      "scoreGenomicLengthLog2scale   -0.25\n",
      "    int: extra score logarithmically scaled with genomic length of the alignment: scoreGenomicLengthLog2scale*log2(genomicLength)\n",
      "\n",
      "scoreDelOpen                 -2\n",
      "    int: deletion open penalty\n",
      "\n",
      "scoreDelBase                 -2\n",
      "    int: deletion extension penalty per base (in addition to scoreDelOpen)\n",
      "\n",
      "scoreInsOpen                 -2\n",
      "    int: insertion open penalty\n",
      "\n",
      "scoreInsBase                 -2\n",
      "    int: insertion extension penalty per base (in addition to scoreInsOpen)\n",
      "\n",
      "scoreStitchSJshift           1\n",
      "    int: maximum score reduction while searching for SJ boundaries in the stitching step\n",
      "\n",
      "\n",
      "### Alignments and Seeding\n",
      "\n",
      "seedSearchStartLmax             50\n",
      "    int>0: defines the search start point through the read - the read is split into pieces no longer than this value\n",
      "\n",
      "seedSearchStartLmaxOverLread    1.0\n",
      "    real: seedSearchStartLmax normalized to read length (sum of mates' lengths for paired-end reads)\n",
      "\n",
      "seedSearchLmax       0\n",
      "    int>=0: defines the maximum length of the seeds, if =0 seed length is not limited\n",
      "\n",
      "seedMultimapNmax      10000\n",
      "    int>0: only pieces that map fewer than this value are utilized in the stitching procedure\n",
      "\n",
      "seedPerReadNmax       1000\n",
      "    int>0: max number of seeds per read\n",
      "\n",
      "seedPerWindowNmax     50\n",
      "    int>0: max number of seeds per window\n",
      "\n",
      "seedNoneLociPerWindow    10\n",
      "    int>0: max number of one seed loci per window\n",
      "\n",
      "seedSplitMin                12\n",
      "    int>0: min length of the seed sequences split by Ns or mate gap\n",
      "\n",
      "seedMapMin              5\n",
      "    int>0: min length of seeds to be mapped\n",
      "\n",
      "alignIntronMin              21\n",
      "    int: minimum intron size, genomic gap is considered intron if its length>=alignIntronMin, otherwise it is considered Deletion\n",
      "\n",
      "alignIntronMax              0\n",
      "    int: maximum intron size, if 0, max intron size will be determined by (2^winBinNbits)*winAnchorDistNbins\n",
      "\n",
      "alignMatesGapMax            0\n",
      "    int: maximum gap between two mates, if 0, max intron gap will be determined by (2^winBinNbits)*winAnchorDistNbins\n",
      "\n",
      "alignSJoverhangMin          5\n",
      "    int>0: minimum overhang (i.e. block size) for spliced alignments\n",
      "\n",
      "alignSJstitchMismatchNmax   0 -1 0 0\n",
      "    4*int>=0: maximum number of mismatches for stitching of the splice junctions (-1: no limit).\n",
      "                            (1) non-canonical motifs, (2) GT/AG and CT/AC motif, (3) GC/AG and CT/GC motif, (4) AT/AC and GT/AT motif.\n",
      "\n",
      "alignSJDBoverhangMin        3\n",
      "    int>0: minimum overhang (i.e. block size) for annotated (sjdb) spliced alignments\n",
      "\n",
      "alignSplicedMateMapLmin     0\n",
      "    int>0: minimum mapped length for a read mate that is spliced\n",
      "\n",
      "alignSplicedMateMapLminOverLmate 0.66\n",
      "    real>0: alignSplicedMateMapLmin normalized to mate length\n",
      "\n",
      "alignWindowsPerReadNmax     10000\n",
      "    int>0: max number of windows per read\n",
      "\n",
      "alignTranscriptsPerWindowNmax       100\n",
      "    int>0: max number of transcripts per window\n",
      "\n",
      "alignTranscriptsPerReadNmax               10000\n",
      "    int>0: max number of different alignments per read to consider\n",
      "\n",
      "alignEndsType           Local\n",
      "    string: type of read ends alignment\n",
      "                        Local             ... standard local alignment with soft-clipping allowed\n",
      "                        EndToEnd          ... force end-to-end read alignment, do not soft-clip\n",
      "                        Extend5pOfRead1   ... fully extend only the 5p of the read1, all other ends: local alignment\n",
      "                        Extend5pOfReads12 ... fully extend only the 5p of the both read1 and read2, all other ends: local alignment\n",
      "\n",
      "alignEndsProtrude       0    ConcordantPair\n",
      "    int, string:        allow protrusion of alignment ends, i.e. start (end) of the +strand mate downstream of the start (end) of the -strand mate\n",
      "                        1st word: int: maximum number of protrusion bases allowed\n",
      "                        2nd word: string:\n",
      "                                            ConcordantPair ... report alignments with non-zero protrusion as concordant pairs\n",
      "                                            DiscordantPair ... report alignments with non-zero protrusion as discordant pairs\n",
      "\n",
      "alignSoftClipAtReferenceEnds    Yes\n",
      "    string: allow the soft-clipping of the alignments past the end of the chromosomes\n",
      "                                Yes ... allow\n",
      "                                No  ... prohibit, useful for compatibility with Cufflinks\n",
      "\n",
      "alignInsertionFlush     None\n",
      "    string: how to flush ambiguous insertion positions\n",
      "                        None    ... insertions are not flushed\n",
      "                        Right   ... insertions are flushed to the right\n",
      "\n",
      "### Paired-End reads\n",
      "peOverlapNbasesMin          0\n",
      "    int>=0:             minimum number of overlapping bases to trigger mates merging and realignment. Specify >0 value to switch on the \"merginf of overlapping mates\" algorithm.\n",
      "\n",
      "peOverlapMMp                0.01\n",
      "    real, >=0 & <1:     maximum proportion of mismatched bases in the overlap area\n",
      "\n",
      "### Windows, Anchors, Binning\n",
      "\n",
      "winAnchorMultimapNmax           50\n",
      "    int>0: max number of loci anchors are allowed to map to\n",
      "\n",
      "winBinNbits                     16\n",
      "    int>0: =log2(winBin), where winBin is the size of the bin for the windows/clustering, each window will occupy an integer number of bins.\n",
      "\n",
      "winAnchorDistNbins              9\n",
      "    int>0: max number of bins between two anchors that allows aggregation of anchors into one window\n",
      "\n",
      "winFlankNbins                   4\n",
      "    int>0: log2(winFlank), where win Flank is the size of the left and right flanking regions for each window\n",
      "\n",
      "winReadCoverageRelativeMin      0.5\n",
      "    real>=0: minimum relative coverage of the read sequence by the seeds in a window, for STARlong algorithm only.\n",
      "\n",
      "winReadCoverageBasesMin      0\n",
      "    int>0: minimum number of bases covered by the seeds in a window , for STARlong algorithm only.\n",
      "\n",
      "### Chimeric Alignments\n",
      "chimOutType                 Junctions\n",
      "    string(s): type of chimeric output\n",
      "                            Junctions       ... Chimeric.out.junction\n",
      "                            SeparateSAMold  ... output old SAM into separate Chimeric.out.sam file\n",
      "                            WithinBAM       ... output into main aligned BAM files (Aligned.*.bam)\n",
      "                            WithinBAM HardClip  ... (default) hard-clipping in the CIGAR for supplemental chimeric alignments (default if no 2nd word is present)\n",
      "                            WithinBAM SoftClip  ... soft-clipping in the CIGAR for supplemental chimeric alignments\n",
      "\n",
      "chimSegmentMin              0\n",
      "    int>=0: minimum length of chimeric segment length, if ==0, no chimeric output\n",
      "\n",
      "chimScoreMin                0\n",
      "    int>=0: minimum total (summed) score of the chimeric segments\n",
      "\n",
      "chimScoreDropMax            20\n",
      "    int>=0: max drop (difference) of chimeric score (the sum of scores of all chimeric segments) from the read length\n",
      "\n",
      "chimScoreSeparation         10\n",
      "    int>=0: minimum difference (separation) between the best chimeric score and the next one\n",
      "\n",
      "chimScoreJunctionNonGTAG    -1\n",
      "    int: penalty for a non-GT/AG chimeric junction\n",
      "\n",
      "chimJunctionOverhangMin     20\n",
      "    int>=0: minimum overhang for a chimeric junction\n",
      "\n",
      "chimSegmentReadGapMax       0\n",
      "    int>=0: maximum gap in the read sequence between chimeric segments\n",
      "\n",
      "chimFilter                  banGenomicN\n",
      "    string(s): different filters for chimeric alignments\n",
      "                            None ... no filtering\n",
      "                            banGenomicN ... Ns are not allowed in the genome sequence around the chimeric junction\n",
      "\n",
      "chimMainSegmentMultNmax        10\n",
      "    int>=1: maximum number of multi-alignments for the main chimeric segment. =1 will prohibit multimapping main segments.\n",
      "\n",
      "chimMultimapNmax                    0\n",
      "    int>=0: maximum number of chimeric multi-alignments\n",
      "                                0 ... use the old scheme for chimeric detection which only considered unique alignments\n",
      "\n",
      "chimMultimapScoreRange          1\n",
      "    int>=0: the score range for multi-mapping chimeras below the best chimeric score. Only works with --chimMultimapNmax > 1\n",
      "\n",
      "chimNonchimScoreDropMin         20\n",
      "    int>=0: to trigger chimeric detection, the drop in the best non-chimeric alignment score with respect to the read length has to be greater than this value\n",
      "\n",
      "chimOutJunctionFormat           0\n",
      "    int: formatting type for the Chimeric.out.junction file\n",
      "                                0 ... no comment lines/headers\n",
      "                                1 ... comment lines at the end of the file: command line and Nreads: total, unique/multi-mapping\n",
      "\n",
      "### Quantification of Annotations\n",
      "quantMode                   -\n",
      "    string(s): types of quantification requested\n",
      "                            -                ... none\n",
      "                            TranscriptomeSAM ... output SAM/BAM alignments to transcriptome into a separate file\n",
      "                            GeneCounts       ... count reads per gene\n",
      "\n",
      "quantTranscriptomeBAMcompression    1\n",
      "    int: -2 to 10  transcriptome BAM compression level\n",
      "                            -2  ... no BAM output\n",
      "                            -1  ... default compression (6?)\n",
      "                             0  ... no compression\n",
      "                             10 ... maximum compression\n",
      "\n",
      "quantTranscriptomeSAMoutput BanSingleEnd_BanIndels_ExtendSoftclip\n",
      "    string: alignment filtering for TranscriptomeSAM output\n",
      "                            BanSingleEnd_BanIndels_ExtendSoftclip ... prohibit indels and single-end alignments, extend softclips - compatible with RSEM\n",
      "                            BanSingleEnd               ... prohibit single-end alignments, allow indels and softclips\n",
      "                            BanSingleEnd_ExtendSoftclip ... prohibit single-end alignments, extend softclips, allow indels\n",
      "\n",
      "\n",
      "### 2-pass Mapping\n",
      "twopassMode                 None\n",
      "    string: 2-pass mapping mode.\n",
      "                            None        ... 1-pass mapping\n",
      "                            Basic       ... basic 2-pass mapping, with all 1st pass junctions inserted into the genome indices on the fly\n",
      "\n",
      "twopass1readsN              -1\n",
      "    int: number of reads to process for the 1st step. Use very large number (or default -1) to map all reads in the first step.\n",
      "\n",
      "\n",
      "### WASP parameters\n",
      "waspOutputMode              None\n",
      "    string: WASP allele-specific output type. This is re-implementation of the original WASP mappability filtering by Bryce van de Geijn, Graham McVicker, Yoav Gilad & Jonathan K Pritchard. Please cite the original WASP paper: Nature Methods 12, 1061–1063 (2015), https://www.nature.com/articles/nmeth.3582 .\n",
      "                            SAMtag      ... add WASP tags to the alignments that pass WASP filtering\n",
      "\n",
      "### STARsolo (single cell RNA-seq) parameters\n",
      "soloType                    None\n",
      "    string(s): type of single-cell RNA-seq\n",
      "                            CB_UMI_Simple   ... (a.k.a. Droplet) one UMI and one Cell Barcode of fixed length in read2, e.g. Drop-seq and 10X Chromium.\n",
      "                            CB_UMI_Complex  ... multiple Cell Barcodes of varying length, one UMI of fixed length and one adapter sequence of fixed length are allowed in read2 only (e.g. inDrop, ddSeq).\n",
      "                            CB_samTagOut    ... output Cell Barcode as CR and/or CB SAm tag. No UMI counting. --readFilesIn cDNA_read1 [cDNA_read2 if paired-end] CellBarcode_read . Requires --outSAMtype BAM Unsorted [and/or SortedByCoordinate]\n",
      "                            SmartSeq        ... Smart-seq: each cell in a separate FASTQ (paired- or single-end), barcodes are corresponding read-groups, no UMI sequences, alignments deduplicated according to alignment start and end (after extending soft-clipped bases)\n",
      "\n",
      "soloCBtype                  Sequence\n",
      "    string: cell barcode type\n",
      "                            Sequence: cell barcode is a sequence (standard option)\n",
      "                            String: cell barcode is an arbitrary string\n",
      "\n",
      "soloCBwhitelist             -\n",
      "    string(s): file(s) with whitelist(s) of cell barcodes. Only --soloType CB_UMI_Complex allows more than one whitelist file.\n",
      "                            None            ... no whitelist: all cell barcodes are allowed\n",
      "\n",
      "soloCBstart                 1\n",
      "    int>0: cell barcode start base\n",
      "\n",
      "soloCBlen                   16\n",
      "    int>0: cell barcode length\n",
      "\n",
      "soloUMIstart                17\n",
      "    int>0: UMI start base\n",
      "\n",
      "soloUMIlen                  10\n",
      "    int>0: UMI length\n",
      "\n",
      "soloBarcodeReadLength       1\n",
      "    int: length of the barcode read\n",
      "                            1   ... equal to sum of soloCBlen+soloUMIlen\n",
      "                            0   ... not defined, do not check\n",
      "\n",
      "soloBarcodeMate             0\n",
      "    int: identifies which read mate contains the barcode (CB+UMI) sequence\n",
      "                            0   ... barcode sequence is on separate read, which should always be the last file in the --readFilesIn listed\n",
      "                            1   ... barcode sequence is a part of mate 1\n",
      "                            2   ... barcode sequence is a part of mate 2\n",
      "\n",
      "soloCBposition              -\n",
      "    strings(s):             position of Cell Barcode(s) on the barcode read.\n",
      "                            Presently only works with --soloType CB_UMI_Complex, and barcodes are assumed to be on Read2.\n",
      "                            Format for each barcode: startAnchor_startPosition_endAnchor_endPosition\n",
      "                            start(end)Anchor defines the Anchor Base for the CB: 0: read start; 1: read end; 2: adapter start; 3: adapter end\n",
      "                            start(end)Position is the 0-based position with of the CB start(end) with respect to the Anchor Base\n",
      "                            String for different barcodes are separated by space.\n",
      "                            Example: inDrop (Zilionis et al, Nat. Protocols, 2017):\n",
      "                            --soloCBposition  0_0_2_-1  3_1_3_8\n",
      "\n",
      "soloUMIposition             -\n",
      "    string:                  position of the UMI on the barcode read, same as soloCBposition\n",
      "                            Example: inDrop (Zilionis et al, Nat. Protocols, 2017):\n",
      "                            --soloCBposition  3_9_3_14\n",
      "\n",
      "soloAdapterSequence         -\n",
      "    string:                 adapter sequence to anchor barcodes. Only one adapter sequence is allowed.\n",
      "\n",
      "soloAdapterMismatchesNmax   1\n",
      "    int>0:                  maximum number of mismatches allowed in adapter sequence.\n",
      "\n",
      "soloCBmatchWLtype           1MM_multi\n",
      "    string:                 matching the Cell Barcodes to the WhiteList\n",
      "                            Exact                           ... only exact matches allowed\n",
      "                            1MM                             ... only one match in whitelist with 1 mismatched base allowed. Allowed CBs have to have at least one read with exact match.\n",
      "                            1MM_multi                       ... multiple matches in whitelist with 1 mismatched base allowed, posterior probability calculation is used choose one of the matches.\n",
      "                                                                Allowed CBs have to have at least one read with exact match. This option matches best with CellRanger 2.2.0\n",
      "                            1MM_multi_pseudocounts          ... same as 1MM_Multi, but pseudocounts of 1 are added to all whitelist barcodes.\n",
      "                            1MM_multi_Nbase_pseudocounts    ... same as 1MM_multi_pseudocounts, multimatching to WL is allowed for CBs with N-bases. This option matches best with CellRanger >= 3.0.0\n",
      "                            EditDist_2                    ... allow up to edit distance of 3 fpr each of the barcodes. May include one deletion + one insertion. Only works with --soloType CB_UMI_Complex. Matches to multiple passlist barcdoes are not allowed. Similar to ParseBio Split-seq pipeline. \n",
      "\n",
      "soloInputSAMattrBarcodeSeq  -\n",
      "    string(s):              when inputting reads from a SAM file (--readsFileType SAM SE/PE), these SAM attributes mark the barcode sequence (in proper order).\n",
      "                            For instance, for 10X CellRanger or STARsolo BAMs, use --soloInputSAMattrBarcodeSeq CR UR .\n",
      "                            This parameter is required when running STARsolo with input from SAM. \n",
      "    \n",
      "soloInputSAMattrBarcodeQual  -\n",
      "    string(s):              when inputting reads from a SAM file (--readsFileType SAM SE/PE), these SAM attributes mark the barcode qualities (in proper order).\n",
      "                            For instance, for 10X CellRanger or STARsolo BAMs, use --soloInputSAMattrBarcodeQual CY UY .\n",
      "                            If this parameter is '-' (default), the quality 'H' will be assigned to all bases.\n",
      "\n",
      "soloStrand                  Forward\n",
      "    string: strandedness of the solo libraries:\n",
      "                            Unstranded  ... no strand information\n",
      "                            Forward     ... read strand same as the original RNA molecule\n",
      "                            Reverse     ... read strand opposite to the original RNA molecule\n",
      "\n",
      "soloFeatures                Gene\n",
      "    string(s): genomic features for which the UMI counts per Cell Barcode are collected\n",
      "                            Gene            ... genes: reads match the gene transcript\n",
      "                            SJ              ... splice junctions: reported in SJ.out.tab\n",
      "                            GeneFull        ... full gene (pre-mRNA): count all reads overlapping genes' exons and introns\n",
      "                            GeneFull_ExonOverIntron ... full gene (pre-mRNA): count all reads overlapping genes' exons and introns: prioritize 100% overlap with exons\n",
      "                            GeneFull_Ex50pAS        ... full gene (pre-RNA): count all reads overlapping genes' exons and introns: prioritize >50% overlap with exons. Do not count reads with 100% exonic overlap in the antisense direction.\n",
      "\n",
      "#####UnderDevelopment_begin : not supported - do not use\n",
      "                            Transcript3p    ... quantification of transcript for 3' protocols\n",
      "#####UnderDevelopment_end\n",
      "\n",
      "soloMultiMappers            Unique\n",
      "    string(s): counting method for reads mapping to multiple genes           \n",
      "                            Unique     ... count only reads that map to unique genes\n",
      "                            Uniform    ... uniformly distribute multi-genic UMIs to all genes\n",
      "                            Rescue     ... distribute UMIs proportionally to unique+uniform counts (~ first iteration of EM)\n",
      "                            PropUnique ... distribute UMIs proportionally to unique mappers, if present, and uniformly if not.\n",
      "                            EM         ... multi-gene UMIs are distributed using Expectation Maximization algorithm\n",
      "\n",
      "soloUMIdedup                1MM_All\n",
      "    string(s):              type of UMI deduplication (collapsing) algorithm\n",
      "                            1MM_All                     ... all UMIs with 1 mismatch distance to each other are collapsed (i.e. counted once).\n",
      "                            1MM_Directional_UMItools    ... follows the \"directional\" method from the UMI-tools by Smith, Heger and Sudbery (Genome Research 2017).\n",
      "                            1MM_Directional             ... same as 1MM_Directional_UMItools, but with more stringent criteria for duplicate UMIs\n",
      "                            Exact                       ... only exactly matching UMIs are collapsed.\n",
      "                            NoDedup                     ... no deduplication of UMIs, count all reads.\n",
      "                            1MM_CR                      ... CellRanger2-4 algorithm for 1MM UMI collapsing.\n",
      "\n",
      "soloUMIfiltering            -\n",
      "    string(s):              type of UMI filtering (for reads uniquely mapping to genes)\n",
      "                            -                  ... basic filtering: remove UMIs with N and homopolymers (similar to CellRanger 2.2.0).\n",
      "                            MultiGeneUMI       ... basic + remove lower-count UMIs that map to more than one gene.\n",
      "                            MultiGeneUMI_All   ... basic + remove all UMIs that map to more than one gene.\n",
      "                            MultiGeneUMI_CR    ... basic + remove lower-count UMIs that map to more than one gene, matching CellRanger > 3.0.0 .\n",
      "                                                   Only works with --soloUMIdedup 1MM_CR\n",
      "                                                \n",
      "soloOutFileNames            Solo.out/          features.tsv barcodes.tsv        matrix.mtx\n",
      "    string(s):              file names for STARsolo output:\n",
      "                            file_name_prefix   gene_names   barcode_sequences   cell_feature_count_matrix\n",
      "\n",
      "soloCellFilter              CellRanger2.2 3000 0.99 10\n",
      "    string(s):              cell filtering type and parameters\n",
      "                            None            ... do not output filtered cells\n",
      "                            TopCells        ... only report top cells by UMI count, followed by the exact number of cells\n",
      "                            CellRanger2.2   ... simple filtering of CellRanger 2.2. \n",
      "                                                Can be followed by numbers: number of expected cells, robust maximum percentile for UMI count, maximum to minimum ratio for UMI count\n",
      "                                                The harcoded values are from CellRanger: nExpectedCells=3000;  maxPercentile=0.99;  maxMinRatio=10\n",
      "                            EmptyDrops_CR   ... EmptyDrops filtering in CellRanger flavor. Please cite the original EmptyDrops paper: A.T.L Lun et al, Genome Biology, 20, 63 (2019): https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1662-y\n",
      "                                                Can be followed by 10 numeric parameters:  nExpectedCells   maxPercentile   maxMinRatio   indMin   indMax   umiMin   umiMinFracMedian   candMaxN   FDR   simN \n",
      "                                                The harcoded values are from CellRanger:             3000            0.99            10    45000    90000      500               0.01      20000  0.01  10000\n",
      "\n",
      "soloOutFormatFeaturesGeneField3    \"Gene Expression\"\n",
      "    string(s):                field 3 in the Gene features.tsv file. If \"-\", then no 3rd field is output.\n",
      "\n",
      "soloCellReadStats           None\n",
      "    string:                 Output reads statistics for each CB\n",
      "                            Standard    ... standard output\n",
      "\n",
      "#####UnderDevelopment_begin : not supported - do not use\n",
      "soloClusterCBfile           -\n",
      "    string:                 file containing the cluster information for cell barcodes, two columns: CB cluster_index. Only used with --soloFeatures Transcript3p\n",
      "#####UnderDevelopment_end\n",
      "(bulk_seq) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "STAR -h"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e170dd96-6e07-4724-ad29-b1bdc3012483",
   "metadata": {},
   "outputs": [],
   "source": [
    "STAR   \n",
    "--runThreadN 16   \n",
    "--genomeDir /local1/PUBLIC_DATA/genomes/refdata-cellranger-arc-GRCh38-2020-A/star/ \n",
    "--readFilesIn /local2/USERS/tomasgomes/bulk_deconv_hsiuchuan/well74_R1.fastq.gz,/local2/USERS/tomasgomes/bulk_deconv_hsiuchuan/well74_R2.fastq.gz      \n",
    "--readFilesCommand zcat      \n",
    "--outFileNamePrefix /local2/USERS/tomasgomes/bulk_deconv_hsiuchuan/aligned/74.   \n",
    "--outSAMtype BAM   SortedByCoordinate      \n",
    "--quantMode GeneCounts\n",
    "\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "50afda34-7168-4378-880f-3e1b353e02f5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ls: cannot access /local2/USERS/tomasgomes/bulk_deconv_hsiuchuan/well74_R1.fastq.gz: No such file or directory\n"
     ]
    },
    {
     "ename": "",
     "evalue": "2",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "ls /local2/USERS/tomasgomes/bulk_deconv_hsiuchuan/well74_R1.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "d3b0ea6d-06e4-41e7-a372-5713915a7afd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[34;42mTMP\u001b[0m  \u001b[01;34mbs-ansible\u001b[0m  \u001b[30;42mscratch\u001b[0m  \u001b[01;34mtest\u001b[0m  \u001b[30;42mtmp\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "ls /local0/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "fa56d3d7-b20d-47a5-b983-f750d362d265",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ls: cannot access /local2/USERS/tomasgomes/bulk_deconv_hsiuchuan/well74_R1.fastq.gz,/local2/USERS/tomasgomes/bulk_deconv_hsiuchuan: No such file or directory\n"
     ]
    },
    {
     "ename": "",
     "evalue": "2",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "ls /local2/USERS/tomasgomes/bulk_deconv_hsiuchuan/well74_R1.fastq.gz,/local2/USERS/tomasgomes/bulk_deconv_hsiuchuan"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "261942e7-95d2-4427-8f1b-1213ef24b7aa",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f636d191-ec3f-4b00-b925-9443a2ce99f1",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "69934ffc-3189-4543-abb8-b1174df417f6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(bulk_seq) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "rm -r iGABA_pre_24_S2_L002_R1_001"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b2a668f6-ad90-428c-b6f8-978b390a0e3f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "e205ab9a-081c-4cc5-b1d4-fee60d36ae72",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;32mUntitled.ipynb\u001b[0m  \u001b[01;34miGABA_pre_24_S1_L001_R1_001\u001b[0m\n",
      "\u001b[01;34miGABA_pre_24\u001b[0m    \u001b[01;34miGABA_pre_24_S2_L002_R1_001\u001b[0m\n",
      "(bulk_seq) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a48c5c4b-b131-4902-a0cf-309e4124bba8",
   "metadata": {},
   "outputs": [],
   "source": [
    "export MAXCORES=8\n",
    "ls 00.RAW_FASTQ/*R1*.fastq.gz | parallel -j 15 --plus --rpl '{/..} s:.fastq.gz::;s:00.RAW_FASTQ/::' \"\n",
    "        mkdir {/..}\n",
    "        cd {/..}\n",
    "       mkdir 00.CLEAN_FQ 01.FASTQC 10.MAPPING 20.MACS2\n",
    "        cd 00.CLEAN_FQ\n",
    "        echo '######## CLEANING ########' >> ../{/..}.log\n",
    "        echo >> ../{/..}.log\n",
    "        echo >> ../{/..}.log\n",
    "        fastq-mcf \\\n",
    "            /staging/leuven/stg_00002/lcb/kdavie/Resources/fastqmcf.adapters \\\n",
    "            ../../00.RAW_FASTQ/{/} \\\n",
    "            -o {/..}.clean.fastq \\\n",
    "            &>> ../{/..}.log \n",
    "       cd ..\n",
    "        echo >> {/..}.log\n",
    "        echo >> {/..}.log\n",
    "        echo '######## QC-ING ########' >> {/..}.log\n",
    "        echo >> {/..}.log\n",
    "        echo >> {/..}.log\n",
    "        fastqc \\\n",
    "        -t ${MAXCORES} 00.CLEAN_FQ/{/..}.clean.fastq \\\n",
    "        -o 01.FASTQC/ \\\n",
    "        &>> {/..}.log\n",
    "        echo >> {/..}.log\n",
    "        echo >> {/..}.log\n",
    "    \n",
    "       STAR \\\n",
    "           --runThreadN ${MAXCORES} \\\n",
    "           --genomeDir /staging/leuven/res_00001/genomes/homo_sapiens/hg38_ucsc/indexes/STAR/2.7.3a \\\n",
    "          --readFilesIn 00.CLEAN_FQ/*.clean.fastq \\\n",
    "          --outFileNamePrefix 10.MAPPING/{/..}_STAR_ \\\n",
    "          --outSAMtype BAM SortedByCoordinate \\\n",
    "          --outWigType bedGraph \\\n",
    "          --outWigStrand Unstranded \\\n",
    "          --outWigNorm RPM \\\n",
    "          --outReadsUnmapped Fastx \\\n",
    "          --quantMode GeneCounts \\\n",
    "          &>> {/..}.log\n",
    "      samtools index 10.MAPPING/{/..}_STAR_Aligned.sortedByCoord.out.bam\n",
    "      echo >> {/..}.log\n",
    "      echo >> {/..}.log\n",
    "   \n",
    "      echo '######## SORTING ########' >> {/..}.log\n",
    "      echo >> {/..}.log\n",
    "      echo >> {/..}.log\n",
    "      samtools view -h -q4 10.MAPPING/{/..}_STAR_Aligned.sortedByCoord.out.bam | samtools sort \\\n",
    "          -@ ${MAXCORES} \\\n",
    "          -O bam \\\n",
    "          -T 10.MAPPING/tmp_sort.bam \\\n",
    "          -o 10.MAPPING/{/..}_q4_sorted.bam \\\n",
    "          - &>> {/..}.log\n",
    "      samtools index 10.MAPPING/{/..}_q4_sorted.bam\n",
    "      echo >> {/..}.log\n",
    "      echo >> {/..}.log\n",
    "      cd ..\n",
    "    \" "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1c55e661-a2b2-4b1b-bc88-6f649c25d517",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "adc80d4c-c850-4c8b-99a5-106b8003f4bb",
   "metadata": {},
   "outputs": [],
   "source": [
    "fastp -i in.R1.fq.gz -I in.R2.fq.gz -o out.R1.fq.gz -O out.R2.fq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e9123456-ce3d-4310-bf6b-197c34593b62",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1d5c87a0-ff96-4d9c-bc34-49174120a905",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fd480a99-952c-4a91-83b6-0ad3f11219c2",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
